Source/Android/jni/MainAndroid.cpp

```diff
@@ -175,13 +175,21 @@ void Host_TitleChanged()
 
 static bool MsgAlert(const char* caption, const char* text, bool yes_no, Common::MsgType style)
 {
-  JNIEnv* env = IDCache::GetEnvForThread();
+  // If a panic alert happens very early in the execution of a game, we can crash here with
+  // the error "JNI NewString called with pending exception java.lang.StackOverflowError".
```

Comment:
Just wondering, where does the java.lang.StackOverflowError come from?

Reply:
I'm not entirely sure. My best guess is that the C++ thread which is generating the panic alert (the CPU thread in my case – not sure if the problem can happen with other threads or not) is using more stack than the Java runtime thinks is okay but not so much that we actually get problems in our C++ code. It's worth noting that the JIT does some "odd" stuff with setting up stack guard pages and putting nonsense in the frame pointer register... though if that causes problems, I don't see why the problems wouldn't also happen when the game already has been running for a while.

Reply:
Yeah, that's just weird... I'm guessing we don't have a useful backtrace to figure out what the PanicAlert is triggered by (and if it's possibly triggered from an unsafe context like in a signal handler), do we?

Reply:
The runtime does spit out a stack trace, but I don't know if you would consider it useful:

<details>
<summary>Stack trace</summary>

```
E/Dolphinemu: 42:01:003 ../../../../../../Core/Common/MsgHandler.cpp:132 E[MASTER]: Warning: Invalid write to 0x00000000, PC = 0x80014968
    42:01:003 ../../../../../../Core/Common/MsgHandler.cpp:132 E[MASTER]: Warning: Invalid write to 0x00000001, PC = 0x80014978
A/olphinemu.debu: java_vm_ext.cc:577] JNI DETECTED ERROR IN APPLICATION: JNI NewString called with pending exception java.lang.StackOverflowError: stack size 995KB
    java_vm_ext.cc:577] (Throwable with empty stack trace)
    java_vm_ext.cc:577] 
    java_vm_ext.cc:577]     in call to NewString
A/olphinemu.debu: runtime.cc:655] Runtime aborting...
    runtime.cc:655] Dumping all threads without mutator lock held
    runtime.cc:655] All threads:
    runtime.cc:655] DALVIK THREADS (43):
    runtime.cc:655] "Thread-12" prio=5 tid=43 Runnable
    runtime.cc:655]   | group="" sCount=0 dsCount=0 flags=0 obj=0x77480000 self=0x720712b840
    runtime.cc:655]   | sysTid=21520 nice=0 cgrp=default sched=0/0 handle=0x700cb2bcc0
    runtime.cc:655]   | state=R schedstat=( 890391960 29124007 351 ) utm=75 stm=13 core=7 HZ=100
    runtime.cc:655]   | stack=0x700ca34000-0x700ca36000 stackSize=995KB
    runtime.cc:655]   | held mutexes= "abort lock" "mutator lock"(shared held)
    runtime.cc:655]   native: #00 pc 000000000049ef64  /apex/com.android.art/lib64/libart.so (art::DumpNativeStack(std::__1::basic_ostream<char, std::__1::char_traits<char> >&, int, BacktraceMap*, char const*, art::ArtMethod*, void*, bool)+140)
    runtime.cc:655]   native: #01 pc 00000000005ac0bc  /apex/com.android.art/lib64/libart.so (art::Thread::DumpStack(std::__1::basic_ostream<char, std::__1::char_traits<char> >&, bool, BacktraceMap*, bool) const+376)
    runtime.cc:655]   native: #02 pc 00000000005c91f4  /apex/com.android.art/lib64/libart.so (art::DumpCheckpoint::Run(art::Thread*)+924)
    runtime.cc:655]   native: #03 pc 00000000005c3134  /apex/com.android.art/lib64/libart.so (art::ThreadList::RunCheckpoint(art::Closure*, art::Closure*)+528)
    runtime.cc:655]   native: #04 pc 00000000005c2300  /apex/com.android.art/lib64/libart.so (art::ThreadList::Dump(std::__1::basic_ostream<char, std::__1::char_traits<char> >&, bool)+1920)
    runtime.cc:655]   native: #05 pc 000000000055cf88  /apex/com.android.art/lib64/libart.so (art::Runtime::Abort(char const*)+1864)
    runtime.cc:655]   native: #06 pc 0000000000013978  /system/lib64/libbase.so (android::base::SetAborter(std::__1::function<void (char const*)>&&)::$_3::__invoke(char const*)+76)
    runtime.cc:655]   native: #07 pc 0000000000012fa4  /system/lib64/libbase.so (android::base::LogMessage::~LogMessage()+320)
    runtime.cc:655]   native: #08 pc 00000000003851a0  /apex/com.android.art/lib64/libart.so (art::JavaVMExt::JniAbort(char const*, char const*)+2572)
    runtime.cc:655]   native: #09 pc 0000000000385218  /apex/com.android.art/lib64/libart.so (art::JavaVMExt::JniAbortV(char const*, char const*, std::__va_list)+108)
    runtime.cc:655]   native: #10 pc 0000000000376efc  /apex/com.android.art/lib64/libart.so (art::(anonymous namespace)::ScopedCheck::AbortF(char const*, ...)+144)
    runtime.cc:655]   native: #11 pc 0000000000375a14  /apex/com.android.art/lib64/libart.so (art::(anonymous namespace)::ScopedCheck::CheckPossibleHeapValue(art::ScopedObjectAccess&, char, art::(anonymous namespace)::JniValueType)+3108)
    runtime.cc:655]   native: #12 pc 0000000000374674  /apex/com.android.art/lib64/libart.so (art::(anonymous namespace)::ScopedCheck::Check(art::ScopedObjectAccess&, bool, char const*, art::(anonymous namespace)::JniValueType*)+612)
    runtime.cc:655]   native: #13 pc 0000000000369af0  /apex/com.android.art/lib64/libart.so (art::(anonymous namespace)::CheckJNI::NewString(_JNIEnv*, unsigned short const*, int)+700)
    runtime.cc:655]   native: #14 pc 0000000000104d04  /data/app/~~EljEbtMxoHPS3CC-1nKN_w==/org.dolphinemu.dolphinemu.debug-MTW6KXGos0F6setl2DoPwQ==/lib/arm64/libmain.so (???)
    runtime.cc:655]   native: #15 pc 0000000000102e74  /data/app/~~EljEbtMxoHPS3CC-1nKN_w==/org.dolphinemu.dolphinemu.debug-MTW6KXGos0F6setl2DoPwQ==/lib/arm64/libmain.so (???)
    runtime.cc:655]   native: #16 pc 00000000001179dc  /data/app/~~EljEbtMxoHPS3CC-1nKN_w==/org.dolphinemu.dolphinemu.debug-MTW6KXGos0F6setl2DoPwQ==/lib/arm64/libmain.so (???)
    runtime.cc:655]   native: #17 pc 00000000001f6430  /data/app/~~EljEbtMxoHPS3CC-1nKN_w==/org.dolphinemu.dolphinemu.debug-MTW6KXGos0F6setl2DoPwQ==/lib/arm64/libmain.so (???)
    runtime.cc:655]   native: #18 pc 0000000002000ea0  <anonymous:6ff9f06000> (???)
    runtime.cc:655]   (no managed stack frames)
    runtime.cc:655] 
    runtime.cc:655] "main" prio=10 tid=1 Native
A/olphinemu.debu: runtime.cc:655]   | group="" sCount=1 dsCount=0 flags=1 obj=0x72fdec68 self=0xb400007206f06380
    runtime.cc:655]   | sysTid=21453 nice=-10 cgrp=default sched=0/0 handle=0x732d9c64f8
    runtime.cc:655]   | state=S schedstat=( 1321520309 90739113 1415 ) utm=118 stm=13 core=4 HZ=100
    runtime.cc:655]   | stack=0x7fd8607000-0x7fd8609000 stackSize=8192KB
    runtime.cc:655]   | held mutexes=
    runtime.cc:655]   native: #00 pc 000000000009ba78  /apex/com.android.runtime/lib64/bionic/libc.so (__epoll_pwait+8)
    runtime.cc:655]   native: #01 pc 0000000000019acc  /system/lib64/libutils.so (android::Looper::pollInner(int)+184)
    runtime.cc:655]   native: #02 pc 00000000000199ac  /system/lib64/libutils.so (android::Looper::pollOnce(int, int*, int*, void**)+112)
    runtime.cc:655]   native: #03 pc 0000000000112b04  /system/lib64/libandroid_runtime.so (android::android_os_MessageQueue_nativePollOnce(_JNIEnv*, _jobject*, long, int)+44)
    runtime.cc:655]   at android.os.MessageQueue.nativePollOnce(Native method)
    runtime.cc:655]   at android.os.MessageQueue.next(MessageQueue.java:335)
    runtime.cc:655]   at android.os.Looper.loop(Looper.java:206)
    runtime.cc:655]   at android.app.ActivityThread.main(ActivityThread.java:7787)
    runtime.cc:655]   at java.lang.reflect.Method.invoke(Native method)
    runtime.cc:655]   at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592)
    runtime.cc:655]   at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:952)
    runtime.cc:655] 
    runtime.cc:655] "Runtime worker thread 1" prio=10 tid=2 Native (still starting up)
    runtime.cc:655]   | group="" sCount=1 dsCount=0 flags=1 obj=0x0 self=0x7206f2c760
    runtime.cc:655]   | sysTid=21460 nice=-20 cgrp=default sched=0/0 handle=0x732a44cd00
    runtime.cc:655]   | state=S schedstat=( 179010 0 3 ) utm=0 stm=0 core=7 HZ=100
    runtime.cc:655]   | stack=0x732a43e000-0x732a440000 stackSize=63KB
    runtime.cc:655]   | held mutexes=
    runtime.cc:655]   native: #00 pc 000000000004adcc  /apex/com.android.runtime/lib64/bionic/libc.so (syscall+28)
    runtime.cc:655]   native: #01 pc 00000000001af924  /apex/com.android.art/lib64/libart.so (art::ConditionVariable::WaitHoldingLocks(art::Thread*)+148)
    runtime.cc:655]   native: #02 pc 00000000005caf1c  /apex/com.android.art/lib64/libart.so (art::ThreadPool::GetTask(art::Thread*)+120)
    runtime.cc:655]   native: #03 pc 00000000005ca180  /apex/com.android.art/lib64/libart.so (art::ThreadPoolWorker::Run()+80)
    runtime.cc:655]   native: #04 pc 00000000005c9c90  /apex/com.android.art/lib64/libart.so (art::ThreadPoolWorker::Callback(void*)+192)
    runtime.cc:655]   native: #05 pc 00000000000af888  /apex/com.android.runtime/lib64/bionic/libc.so (__pthread_start(void*)+64)
    runtime.cc:655]   native: #06 pc 000000000004fd08  /apex/com.android.runtime/lib64/bionic/libc.so (__start_thread+64)
    runtime.cc:655]   (no managed stack frames)
    runtime.cc:655] 
    runtime.cc:655] "Runtime worker thread 2" prio=10 tid=3 Native (still starting up)
    runtime.cc:655]   | group="" sCount=1 dsCount=0 flags=1 obj=0x0 self=0x7206f28fc0
    runtime.cc:655]   | sysTid=21461 nice=-20 cgrp=default sched=0/0 handle=0x73270cad00
    runtime.cc:655]   | state=S schedstat=( 136770 0 6 ) utm=0 stm=0 core=7 HZ=100
    runtime.cc:655]   | stack=0x73270bc000-0x73270be000 stackSize=63KB
    runtime.cc:655]   | held mutexes=
    runtime.cc:655]   native: #00 pc 000000000004adcc  /apex/com.android.runtime/lib64/bionic/libc.so (syscall+28)
    runtime.cc:655]   native: #01 pc 00000000001af924  /apex/com.android.art/lib64/libart.so (art::ConditionVariable::WaitHoldingLocks(art::Thread*)+148)
    runtime.cc:655]   native: #02 pc 00000000005caf1c  /apex/com.android.art/lib64/libart.so (art::ThreadPool::GetTask(art::Thread*)+120)
    runtime.cc:655]   native: #03 pc 00000000005ca180  /apex/com.android.art/lib64/libart.so (art::ThreadPoolWorker::Run()+80)
    runtime.cc:655]   native: #04 pc 00000000005c9c90  /apex/com.android.art/lib64/libart.so (art::ThreadPoolWorker::Callback(void*)+192)
    runtime.cc:655]   native: #05 pc 00000000000af888  /apex/com.android.runtime/lib64/bionic/libc.so (__pthread_start(void*)+64)
    runtime.cc:655]   native: #06 pc 000000000004fd08  /apex/com.android.runtime/lib64/bionic/libc.so (__start_thread+64)
    runtime.cc:655]   (no managed stack frames)
    runtime.cc:655] 
    runtime.cc:655] "Runtime worker thread 3" prio=10 tid=4 Native (still starting up)
    runtime.cc:655]   | group="" sCount=1 dsCount=0 flags=1 obj=0x0 self=0x7206f09b20
    runtime.cc:655]   | sysTid=21462 nice=-20 cgrp=default sched=0/0 handle=0x70967ebd00
    runtime.cc:655]   | state=S schedstat=( 50260 0 1 ) utm=0 stm=0 core=5 HZ=100
    runtime.cc:655]   | stack=0x70967dd000-0x70967df000 stackSize=63KB
    runtime.cc:655]   | held mutexes=
    runtime.cc:655]   native: #00 pc 000000000004adcc  /apex/com.android.runtime/lib64/bionic/libc.so (syscall+28)
    runtime.cc:655]   native: #01 pc 00000000001af924  /apex/com.android.art/lib64/libart.so (art::ConditionVariable::WaitHoldingLocks(art::Thread*)+148)
    runtime.cc:655]   native: #02 pc 00000000005caf1c  /apex/com.android.art/lib64/libart.so (art::ThreadPool::GetTask(art::Thread*)+120)
    runtime.cc:655]   native: #03 pc 00000000005ca180  /apex/com.android.art/lib64/libart.so (art::ThreadPoolWorker::Run()+80)
    runtime.cc:655]   native: #04 pc 00000000005c9c90  /apex/com.android.art/lib64/libart.so (art::ThreadPoolWorker::Callback(void*)+192)
    runtime.cc:655]   native: #05 pc 00000000000af888  /apex/com.android.runtime/lib64/bionic/libc.so (__pthread_start(void*)+64)
    runtime.cc:655]   native: #06 pc 000000000004fd08  /apex/com.android.runtime/lib64/bionic/libc.so (__start_thread+64)
    runtime.cc:655]   (no managed stack frames)
    runtime.cc:655] 
    runtime.cc:655] "Signal Catcher" prio=10 tid=5 WaitingInMainSignalCatcherLoop
    runtime.cc:655]   | group="" sCount=1 dsCount=0 flags=1 obj=0x79400250 self=0x7206f23c50
    runtime.cc:655]   | sysTid=21463 nice=-20 cgrp=default sched=0/0 handle=0x7084519cc0
    runtime.cc:655]   | state=S schedstat=( 277812 0 1 ) utm=0 stm=0 core=5 HZ=100
    runtime.cc:655]   | stack=0x7084422000-0x7084424000 stackSize=995KB
    runtime.cc:655]   | held mutexes=
    runtime.cc:655]   native: #00 pc 000000000009b4b8  /apex/com.android.runtime/lib64/bionic/libc.so (__rt_sigtimedwait+8)
    runtime.cc:655]   native: #01 pc 000000000005c5fc  /apex/com.android.runtime/lib64/bionic/libc.so (sigwait+68)
    runtime.cc:655]   native: #02 pc 000000000058307c  /apex/com.android.art/lib64/libart.so (art::SignalCatcher::WaitForSignal(art::Thread*, art::SignalSet&)+428)
    runtime.cc:655]   native: #03 pc 0000000000581c74  /apex/com.android.art/lib64/libart.so (art::SignalCatcher::Run(void*)+272)
    runtime.cc:655]   native: #04 pc 00000000000af888  /apex/com.android.runtime/lib64/bionic/libc.so (__pthread_start(void*)+64)
    runtime.cc:655]   native: #05 pc 000000000004fd08  /apex/com.android.runtime/lib64/bionic/libc.so (__start_thread+64)
    runtime.cc:655]   (no managed stack frames)
    runtime.cc:655] 
    runtime.cc:655] "Runtime worker thread 0" prio=10 tid=6 Native (still starting up)
    runtime.cc:655]   | group="" sCount=1 dsCount=0 flags=1 obj=0x0 self=0x7206f1e8e0
    runtime.cc:655]   | sysTid=21459 nice=-20 cgrp=default sched=0/0 handle=0x732c50cd00
    runtime.cc:655]   | state=S schedstat=( 139375 1342813 1 ) utm=0 stm=0 core=6 HZ=100
    runtime.cc:655]   | stack=0x732c4fe000-0x732c500000 stackSize=63KB
    runtime.cc:655]   | held mutexes=
    runtime.cc:655]   native: #00 pc 000000000004adcc  /apex/com.android.runtime/lib64/bionic/libc.so (syscall+28)
    runtime.cc:655]   native: #01 pc 00000000001af924  /apex/com.android.art/lib64/libart.so (art::ConditionVariable::WaitHoldingLocks(art::Thread*)+148)
    runtime.cc:655]   native: #02 pc 00000000005caf1c  /apex/com.android.art/lib64/libart.so (art::ThreadPool::GetTask(art::Thread*)+120)
    runtime.cc:655]   native: #03 pc 00000000005ca180  /apex/com.android.art/lib64/libart.so (art::ThreadPoolWorker::Run()+80)
    runtime.cc:655]   native: #04 pc 00000000005c9c90  /apex/com.android.art/lib64/libart.so (art::ThreadPoolWorker::Callback(void*)+192)
    runtime.cc:655]   native: #05 pc 00000000000af888  /apex/com.android.runtime/lib64/bionic/libc.so (__pthread_start(void*)+64)
    runtime.cc:655]   native: #06 pc 000000000004fd08  /apex/com.android.runtime/lib64/bionic/libc.so (__start_thread+64)
    runtime.cc:655]   (no managed stack frames)
    runtime.cc:655] 
    runtime.cc:655] "perfetto_hprof_listener" prio=10 tid=7 Native (still starting up)
    runtime.cc:655]   | group="" sCount=1 dsCount=0 flags=1 obj=0x0 self=0x7206f0d2c0
    runtime.cc:655]   | sysTid=21464 nice=-20 cgrp=default sched=0/0 handle=0x70823b7cc0
    runtime.cc:655]   | state=S schedstat=( 176614 0 7 ) utm=0 stm=0 core=5 HZ=100
    runtime.cc:655]   | stack=0x70822c0000-0x70822c2000 stackSize=995KB
    runtime.cc:655]   | held mutexes=
    runtime.cc:655]   native: #00 pc 000000000009a834  /apex/com.android.runtime/lib64/bionic/libc.so (read+4)
    runtime.cc:655]   native: #01 pc 00000000000180bc  /apex/com.android.art/lib64/libperfetto_hprof.so (void* std::__1::__thread_proxy<std::__1::tuple<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct> >, ArtPlugin_Initialize::$_29> >(void*)+280)
    runtime.cc:655]   native: #02 pc 00000000000af888  /apex/com.android.runtime/lib64/bionic/libc.so (__pthread_start(void*)+64)
    runtime.cc:655]   native: #03 pc 000000000004fd08  /apex/com.android.runtime/lib64/bionic/libc.so (__start_thread+64)
    runtime.cc:655]   (no managed stack frames)
    runtime.cc:655] 
    runtime.cc:655] "ADB-JDWP Connection Control Thread" prio=10 tid=8 WaitingInMainDebuggerLoop
    runtime.cc:655]   | group="" sCount=1 dsCount=0 flags=1 obj=0x794002c8 self=0x7206f10a60
    runtime.cc:655]   | sysTid=21465 nice=-20 cgrp=default sched=0/0 handle=0x70362fdcc0
    runtime.cc:655]   | state=S schedstat=( 1117761 10156 10 ) utm=0 stm=0 core=5 HZ=100
    runtime.cc:655]   | stack=0x7036206000-0x7036208000 stackSize=995KB
    runtime.cc:655]   | held mutexes=
    runtime.cc:655]   native: #00 pc 000000000009bb78  /apex/com.android.runtime/lib64/bionic/libc.so (__ppoll+8)
    runtime.cc:655]   native: #01 pc 0000000000059bc4  /apex/com.android.runtime/lib64/bionic/libc.so (poll+92)
    runtime.cc:655]   native: #02 pc 0000000000009efc  /apex/com.android.art/lib64/libadbconnection.so (adbconnection::AdbConnectionState::RunPollLoop(art::Thread*)+828)
    runtime.cc:655]   native: #03 pc 0000000000008578  /apex/com.android.art/lib64/libadbconnection.so (adbconnection::CallbackFunction(void*)+1508)
    runtime.cc:655]   native: #04 pc 00000000000af888  /apex/com.android.runtime/lib64/bionic/libc.so (__pthread_start(void*)+64)
    runtime.cc:655]   native: #05 pc 000000000004fd08  /apex/com.android.runtime/lib64/bionic/libc.so (__start_thread+64)
    runtime.cc:655]   (no managed stack frames)
    runtime.cc:655] 
    runtime.cc:655] "HeapTaskDaemon" prio=5 tid=9 WaitingForTaskProcessor
    runtime.cc:655]   | group="" sCount=1 dsCount=0 flags=1 obj=0x79405cc0 self=0x7206f0b6f0
    runtime.cc:655]   | sysTid=21467 nice=4 cgrp=default sched=0/0 handle=0x70350fecc0
    runtime.cc:655]   | state=S schedstat=( 20095679 730000 7 ) utm=2 stm=0 core=5 HZ=100
    runtime.cc:655]   | stack=0x7034ffb000-0x7034ffd000 stackSize=1043KB
    runtime.cc:655]   | held mutexes=
    runtime.cc:655]   native: #00 pc 000000000004add0  /apex/com.android.runtime/lib64/bionic/libc.so (syscall+32)
    runtime.cc:655]   native: #01 pc 00000000001afcf4  /apex/com.android.art/lib64/libart.so (art::ConditionVariable::TimedWait(art::Thread*, long, int)+172)
    runtime.cc:655]   native: #02 pc 00000000002e52bc  /apex/com.android.art/lib64/libart.so (art::gc::TaskProcessor::GetTask(art::Thread*)+512)
    runtime.cc:655]   native: #03 pc 00000000002e5c04  /apex/com.android.art/lib64/libart.so (art::gc::TaskProcessor::RunAllTasks(art::Thread*)+92)
    runtime.cc:655]   at dalvik.system.VMRuntime.runHeapTasks(Native method)
    runtime.cc:655]   at java.lang.Daemons$HeapTaskDaemon.runInternal(Daemons.java:531)
    runtime.cc:655]   at java.lang.Daemons$Daemon.run(Daemons.java:139)
    runtime.cc:655]   at java.lang.Thread.run(Thread.java:923)
    runtime.cc:655] 
    runtime.cc:655] "Jit thread pool worker thread 0" prio=5 tid=10 Native
    runtime.cc:655]   | group="" sCount=1 dsCount=0 flags=1 obj=0x79400340 self=0x7206f07f50
    runtime.cc:655]   | sysTid=21466 nice=0 cgrp=default sched=0/0 handle=0x7035203d00
    runtime.cc:655]   | state=S schedstat=( 168900108 48972711 393 ) utm=13 stm=3 core=4 HZ=100
    runtime.cc:655]   | stack=0x7035105000-0x7035107000 stackSize=1023KB
    runtime.cc:655]   | held mutexes=
    runtime.cc:655]   native: #00 pc 000000000004adcc  /apex/com.android.runtime/lib64/bionic/libc.so (syscall+28)
    runtime.cc:655]   native: #01 pc 00000000001af924  /apex/com.android.art/lib64/libart.so (art::ConditionVariable::WaitHoldingLocks(art::Thread*)+148)
    runtime.cc:655]   native: #02 pc 00000000005caf1c  /apex/com.android.art/lib64/libart.so (art::ThreadPool::GetTask(art::Thread*)+120)
    runtime.cc:655]   native: #03 pc 00000000005ca1c0  /apex/com.android.art/lib64/libart.so (art::ThreadPoolWorker::Run()+144)
    runtime.cc:655]   native: #04 pc 00000000005c9c90  /apex/com.android.art/lib64/libart.so (art::ThreadPoolWorker::Callback(void*)+192)
    runtime.cc:655]   native: #05 pc 00000000000af888  /apex/com.android.runtime/lib64/bionic/libc.so (__pthread_start(void*)+64)
    runtime.cc:655]   native: #06 pc 000000000004fd08  /apex/com.android.runtime/lib64/bionic/libc.so (__start_thread+64)
    runtime.cc:655]   (no managed stack frames)
    runtime.cc:655] 
    runtime.cc:655] "ReferenceQueueDaemon" prio=5 tid=11 Waiting
    runtime.cc:655]   | group="" sCount=1 dsCount=0 flags=1 obj=0x794003b8 self=0x7206f1b140
    runtime.cc:655]   | sysTid=21468 nice=4 cgrp=default sched=0/0 handle=0x7033ff4cc0
    runtime.cc:655]   | state=S schedstat=( 1394166 0 5 ) utm=0 stm=0 core=4 HZ=100
    runtime.cc:655]   | stack=0x7033ef1000-0x7033ef3000 stackSize=1043KB
    runtime.cc:655]   | held mutexes=
    runtime.cc:655]   native: #00 pc 000000000004adcc  /apex/com.android.runtime/lib64/bionic/libc.so (syscall+28)
    runtime.cc:655]   native: #01 pc 00000000001af924  /apex/com.android.art/lib64/libart.so (art::ConditionVariable::WaitHoldingLocks(art::Thread*)+148)
    runtime.cc:655]   native: #02 pc 000000000049a160  /apex/com.android.art/lib64/libart.so (art::Monitor::Wait(art::Thread*, long, int, bool, art::ThreadState)+544)
    runtime.cc:655]   native: #03 pc 000000000049b9b4  /apex/com.android.art/lib64/libart.so (art::Monitor::Wait(art::Thread*, art::ObjPtr<art::mirror::Object>, long, int, bool, art::ThreadState)+280)
    runtime.cc:655]   at java.lang.Object.wait(Native method)
    runtime.cc:655]   - waiting on <0x09a44b2d> (a java.lang.Class<java.lang.ref.ReferenceQueue>)
    runtime.cc:655]   at java.lang.Object.wait(Object.java:442)
    runtime.cc:655]   at java.lang.Object.wait(Object.java:568)
    runtime.cc:655]   at java.lang.Daemons$ReferenceQueueDaemon.runInternal(Daemons.java:217)
    runtime.cc:655]   - locked <0x09a44b2d> (a java.lang.Class<java.lang.ref.ReferenceQueue>)
    runtime.cc:655]   at java.lang.Daemons$Daemon.run(Daemons.java:139)
    runtime.cc:655]   at java.lang.Thread.run(Thread.java:923)
    runtime.cc:655] 
    runtime.cc:655] "FinalizerDaemon" prio=5 tid=12 Waiting
    runtime.cc:655]   | group="" sCount=1 dsCount=0 flags=1 obj=0x79400430 self=0x7206f15dd0
    runtime.cc:655]   | sysTid=21469 nice=4 cgrp=default sched=0/0 handle=0x7031eeacc0
    runtime.cc:655]   | state=S schedstat=( 1368386 116510 7 ) utm=0 stm=0 core=6 HZ=100
    runtime.cc:655]   | stack=0x7031de7000-0x7031de9000 stackSize=1043KB
    runtime.cc:655]   | held mutexes=
    runtime.cc:655]   native: #00 pc 000000000004adcc  /apex/com.android.runtime/lib64/bionic/libc.so (syscall+28)
    runtime.cc:655]   native: #01 pc 00000000001af924  /apex/com.android.art/lib64/libart.so (art::ConditionVariable::WaitHoldingLocks(art::Thread*)+148)
    runtime.cc:655]   native: #02 pc 000000000049a160  /apex/com.android.art/lib64/libart.so (art::Monitor::Wait(art::Thread*, long, int, bool, art::ThreadState)+544)
    runtime.cc:655]   native: #03 pc 000000000049b9b4  /apex/com.android.art/lib64/libart.so (art::Monitor::Wait(art::Thread*, art::ObjPtr<art::mirror::Object>, long, int, bool, art::ThreadState)+280)
    runtime.cc:655]   at java.lang.Object.wait(Native method)
    runtime.cc:655]   - waiting on <0x0ba9e862> (a java.lang.Object)
    runtime.cc:655]   at java.lang.Object.wait(Object.java:442)
    runtime.cc:655]   at java.lang.ref.ReferenceQueue.remove(ReferenceQueue.java:190)
    runtime.cc:655]   - locked <0x0ba9e862> (a java.lang.Object)
    runtime.cc:655]   at java.lang.ref.ReferenceQueue.remove(ReferenceQueue.java:211)
    runtime.cc:655]   at java.lang.Daemons$FinalizerDaemon.runInternal(Daemons.java:273)
    runtime.cc:655]   at java.lang.Daemons$Daemon.run(Daemons.java:139)
    runtime.cc:655]   at java.lang.Thread.run(Thread.java:923)
    runtime.cc:655] 
    runtime.cc:655] "FinalizerWatchdogDaemon" prio=5 tid=13 Sleeping
    runtime.cc:655]   | group="" sCount=1 dsCount=0 flags=1 obj=0x794004a8 self=0x7206f02be0
    runtime.cc:655]   | sysTid=21470 nice=4 cgrp=default sched=0/0 handle=0x7031de0cc0
    runtime.cc:655]   | state=S schedstat=( 204738 80313 7 ) utm=0 stm=0 core=5 HZ=100
    runtime.cc:655]   | stack=0x7031cdd000-0x7031cdf000 stackSize=1043KB
    runtime.cc:655]   | held mutexes=
    runtime.cc:655]   native: #00 pc 000000000004add0  /apex/com.android.runtime/lib64/bionic/libc.so (syscall+32)
    runtime.cc:655]   native: #01 pc 00000000001afcf4  /apex/com.android.art/lib64/libart.so (art::ConditionVariable::TimedWait(art::Thread*, long, int)+172)
    runtime.cc:655]   native: #02 pc 000000000049a174  /apex/com.android.art/lib64/libart.so (art::Monitor::Wait(art::Thread*, long, int, bool, art::ThreadState)+564)
    runtime.cc:655]   native: #03 pc 000000000049b9b4  /apex/com.android.art/lib64/libart.so (art::Monitor::Wait(art::Thread*, art::ObjPtr<art::mirror::Object>, long, int, bool, art::ThreadState)+280)
    runtime.cc:655]   native: #04 pc 000000000013ced4  /apex/com.android.art/lib64/libart.so (art_quick_generic_jni_trampoline+148)
    runtime.cc:655]   native: #05 pc 00000000001337e8  /apex/com.android.art/lib64/libart.so (art_quick_invoke_static_stub+568)
    runtime.cc:655]   native: #06 pc 00000000001a8a94  /apex/com.android.art/lib64/libart.so (art::ArtMethod::Invoke(art::Thread*, unsigned int*, unsigned int, art::JValue*, char const*)+228)
    runtime.cc:655]   native: #07 pc 0000000000318304  /apex/com.android.art/lib64/libart.so (art::interpreter::ArtInterpreterToCompiledCodeBridge(art::Thread*, art::ArtMethod*, art::ShadowFrame*, unsigned short, art::JValue*)+376)
    runtime.cc:655]   native: #08 pc 000000000030e630  /apex/com.android.art/lib64/libart.so (bool art::interpreter::DoCall<false, false>(art::ArtMethod*, art::Thread*, art::ShadowFrame&, art::Instruction const*, unsigned short, art::JValue*)+996)
    runtime.cc:655]   native: #09 pc 000000000067f514  /apex/com.android.art/lib64/libart.so (MterpInvokeStatic+548)
    runtime.cc:655]   native: #10 pc 000000000012d994  /apex/com.android.art/lib64/libart.so (mterp_op_invoke_static+20)
    runtime.cc:655]   native: #11 pc 0000000000002e2a  [anon:dalvik-/apex/com.android.art/javalib/core-oj.jar-transformed] (java.lang.Thread.sleep+118)
    runtime.cc:655]   native: #12 pc 000000000067f7b8  /apex/com.android.art/lib64/libart.so (MterpInvokeStatic+1224)
    runtime.cc:655]   native: #13 pc 000000000012d994  /apex/com.android.art/lib64/libart.so (mterp_op_invoke_static+20)
    runtime.cc:655]   native: #14 pc 0000000000002d9a  [anon:dalvik-/apex/com.android.art/javalib/core-oj.jar-transformed] (java.lang.Thread.sleep+2)
    runtime.cc:655]   native: #15 pc 000000000067f7b8  /apex/com.android.art/lib64/libart.so (MterpInvokeStatic+1224)
    runtime.cc:655]   native: #16 pc 000000000012d994  /apex/com.android.art/lib64/libart.so (mterp_op_invoke_static+20)
    runtime.cc:655]   native: #17 pc 000000000002d1da  /apex/com.android.art/javalib/core-libart.jar (java.lang.Daemons$FinalizerWatchdogDaemon.sleepForNanos+58)
    runtime.cc:655]   native: #18 pc 000000000067eed8  /apex/com.android.art/lib64/libart.so (MterpInvokeDirect+1248)
    runtime.cc:655]   native: #19 pc 000000000012d914  /apex/com.android.art/lib64/libart.so (mterp_op_invoke_direct+20)
    runtime.cc:655]   native: #20 pc 000000000002d2e8  /apex/com.android.art/javalib/core-libart.jar (java.lang.Daemons$FinalizerWatchdogDaemon.waitForFinalization+80)
    runtime.cc:655]   native: #21 pc 000000000067eed8  /apex/com.android.art/lib64/libart.so (MterpInvokeDirect+1248)
    runtime.cc:655]   native: #22 pc 000000000012d914  /apex/com.android.art/lib64/libart.so (mterp_op_invoke_direct+20)
    runtime.cc:655]   native: #23 pc 000000000002d54a  /apex/com.android.art/javalib/core-libart.jar (java.lang.Daemons$FinalizerWatchdogDaemon.runInternal+26)
    runtime.cc:655]   native: #24 pc 000000000067c85c  /apex/com.android.art/lib64/libart.so (MterpInvokeVirtual+1520)
    runtime.cc:655]   native: #25 pc 000000000012d814  /apex/com.android.art/lib64/libart.so (mterp_op_invoke_virtual+20)
    runtime.cc:655]   native: #26 pc 000000000002ce46  /apex/com.android.art/javalib/core-libart.jar (java.lang.Daemons$Daemon.run+50)
    runtime.cc:655]   native: #27 pc 000000000067e294  /apex/com.android.art/lib64/libart.so (MterpInvokeInterface+1808)
    runtime.cc:655]   native: #28 pc 000000000012da14  /apex/com.android.art/lib64/libart.so (mterp_op_invoke_interface+20)
    runtime.cc:655]   native: #29 pc 00000000000eb840  /apex/com.android.art/javalib/core-oj.jar (java.lang.Thread.run+8)
    runtime.cc:655]   native: #30 pc 0000000000305c2c  /apex/com.android.art/lib64/libart.so (art::interpreter::Execute(art::Thread*, art::CodeItemDataAccessor const&, art::ShadowFrame&, art::JValue, bool, bool) (.llvm.17742421990365963658)+268)
    runtime.cc:655]   native: #31 pc 000000000066b2c4  /apex/com.android.art/lib64/libart.so (artQuickToInterpreterBridge+780)
    runtime.cc:655]   native: #32 pc 000000000013cff8  /apex/com.android.art/lib64/libart.so (art_quick_to_interpreter_bridge+88)
    runtime.cc:655]   native: #33 pc 0000000000133564  /apex/com.android.art/lib64/libart.so (art_quick_invoke_stub+548)
    runtime.cc:655]   native: #34 pc 00000000001a8a78  /apex/com.android.art/lib64/libart.so (art::ArtMethod::Invoke(art::Thread*, unsigned int*, unsigned int, art::JValue*, char const*)+200)
    runtime.cc:655]   native: #35 pc 0000000000554d80  /apex/com.android.art/lib64/libart.so (art::JValue art::InvokeVirtualOrInterfaceWithJValues<art::ArtMethod*>(art::ScopedObjectAccessAlreadyRunnable const&, _jobject*, art::ArtMethod*, jvalue const*)+460)
    runtime.cc:655]   native: #36 pc 00000000005a411c  /apex/com.android.art/lib64/libart.so (art::Thread::CreateCallback(void*)+1308)
    runtime.cc:655]   native: #37 pc 00000000000af888  /apex/com.android.runtime/lib64/bionic/libc.so (__pthread_start(void*)+64)
    runtime.cc:655]   native: #38 pc 000000000004fd08  /apex/com.android.runtime/lib64/bionic/libc.so (__start_thread+64)
    runtime.cc:655]   at java.lang.Thread.sleep(Native method)
    runtime.cc:655]   - sleeping on <0x09f1fdf3> (a java.lang.Object)
    runtime.cc:655]   at java.lang.Thread.sleep(Thread.java:442)
    runtime.cc:655]   - locked <0x09f1fdf3> (a java.lang.Object)
    runtime.cc:655]   at java.lang.Thread.sleep(Thread.java:358)
    runtime.cc:655]   at java.lang.Daemons$FinalizerWatchdogDaemon.sleepForNanos(Daemons.java:390)
    runtime.cc:655]   at java.lang.Daemons$FinalizerWatchdogDaemon.waitForFinalization(Daemons.java:419)
    runtime.cc:655]   at java.lang.Daemons$FinalizerWatchdogDaemon.runInternal(Daemons.java:325)
    runtime.cc:655]   at java.lang.Daemons$Daemon.run(Daemons.java:139)
    runtime.cc:655]   at java.lang.Thread.run(Thread.java:923)
    runtime.cc:655] 
    runtime.cc:655] "Binder:21453_1" prio=5 tid=14 Native
    runtime.cc:655]   | group="" sCount=1 dsCount=0 flags=1 obj=0x79400520 self=0x7206f01010
    runtime.cc:655]   | sysTid=21471 nice=0 cgrp=default sched=0/0 handle=0x702fbb8cc0
    runtime.cc:655]   | state=S schedstat=( 1059689 2243282 11 ) utm=0 stm=0 core=4 HZ=100
    runtime.cc:655]   | stack=0x702fac1000-0x702fac3000 stackSize=995KB
    runtime.cc:655]   | held mutexes=
    runtime.cc:655]   native: #00 pc 000000000009aab4  /apex/com.android.runtime/lib64/bionic/libc.so (__ioctl+4)
    runtime.cc:655]   native: #01 pc 00000000000575e4  /apex/com.android.runtime/lib64/bionic/libc.so (ioctl+156)
    runtime.cc:655]   native: #02 pc 00000000000519e4  /system/lib64/libbinder.so (android::IPCThreadState::talkWithDriver(bool)+296)
    runtime.cc:655]   native: #03 pc 0000000000051bd4  /system/lib64/libbinder.so (android::IPCThreadState::getAndExecuteCommand()+24)
    runtime.cc:655]   native: #04 pc 0000000000052490  /system/lib64/libbinder.so (android::IPCThreadState::joinThreadPool(bool)+60)
    runtime.cc:655]   native: #05 pc 0000000000078470  /system/lib64/libbinder.so (android::PoolThread::threadLoop()+24)
    runtime.cc:655]   native: #06 pc 00000000000154cc  /system/lib64/libutils.so (android::Thread::_threadLoop(void*)+260)
    runtime.cc:655]   native: #07 pc 00000000000a2654  /system/lib64/libandroid_runtime.so (android::AndroidRuntime::javaThreadShell(void*)+144)
    runtime.cc:655]   native: #08 pc 0000000000014d90  /system/lib64/libutils.so (thread_data_t::trampoline(thread_data_t const*)+412)
    runtime.cc:655]   native: #09 pc 00000000000af888  /apex/com.android.runtime/lib64/bionic/libc.so (__pthread_start(void*)+64)
    runtime.cc:655]   native: #10 pc 000000000004fd08  /apex/com.android.runtime/lib64/bionic/libc.so (__start_thread+64)
    runtime.cc:655]   (no managed stack frames)
    runtime.cc:655] 
    runtime.cc:655] "Binder:21453_2" prio=5 tid=15 Native
    runtime.cc:655]   | group="" sCount=1 dsCount=0 flags=1 obj=0x79400598 self=0x7206f1cd10
    runtime.cc:655]   | sysTid=21472 nice=0 cgrp=default sched=0/0 handle=0x702eabacc0
    runtime.cc:655]   | state=S schedstat=( 13114116 5438802 33 ) utm=1 stm=0 core=6 HZ=100
    runtime.cc:655]   | stack=0x702e9c3000-0x702e9c5000 stackSize=995KB
    runtime.cc:655]   | held mutexes=
    runtime.cc:655]   native: #00 pc 000000000009aab4  /apex/com.android.runtime/lib64/bionic/libc.so (__ioctl+4)
    runtime.cc:655]   native: #01 pc 00000000000575e4  /apex/com.android.runtime/lib64/bionic/libc.so (ioctl+156)
    runtime.cc:655]   native: #02 pc 00000000000519e4  /system/lib64/libbinder.so (android::IPCThreadState::talkWithDriver(bool)+296)
    runtime.cc:655]   native: #03 pc 0000000000051bd4  /system/lib64/libbinder.so (android::IPCThreadState::getAndExecuteCommand()+24)
    runtime.cc:655]   native: #04 pc 0000000000052490  /system/lib64/libbinder.so (android::IPCThreadState::joinThreadPool(bool)+60)
    runtime.cc:655]   native: #05 pc 0000000000078470  /system/lib64/libbinder.so (android::PoolThread::threadLoop()+24)
    runtime.cc:655]   native: #06 pc 00000000000154cc  /system/lib64/libutils.so (android::Thread::_threadLoop(void*)+260)
    runtime.cc:655]   native: #07 pc 00000000000a2654  /system/lib64/libandroid_runtime.so (android::AndroidRuntime::javaThreadShell(void*)+144)
    runtime.cc:655]   native: #08 pc 0000000000014d90  /system/lib64/libutils.so (thread_data_t::trampoline(thread_data_t const*)+412)
    runtime.cc:655]   native: #09 pc 00000000000af888  /apex/com.android.runtime/lib64/bionic/libc.so (__pthread_start(void*)+64)
    runtime.cc:655]   native: #10 pc 000000000004fd08  /apex/com.android.runtime/lib64/bionic/libc.so (__start_thread+64)
    runtime.cc:655]   (no managed stack frames)
    runtime.cc:655] 
    runtime.cc:655] "Binder:21453_3" prio=5 tid=16 Native
    runtime.cc:655]   | group="" sCount=1 dsCount=0 flags=1 obj=0x79400610 self=0x7206f273f0
    runtime.cc:655]   | sysTid=21473 nice=0 cgrp=default sched=0/0 handle=0x702d9bccc0
    runtime.cc:655]   | state=S schedstat=( 6073701 7429270 25 ) utm=0 stm=0 core=6 HZ=100
    runtime.cc:655]   | stack=0x702d8c5000-0x702d8c7000 stackSize=995KB
    runtime.cc:655]   | held mutexes=
    runtime.cc:655]   native: #00 pc 000000000009aab4  /apex/com.android.runtime/lib64/bionic/libc.so (__ioctl+4)
    runtime.cc:655]   native: #01 pc 00000000000575e4  /apex/com.android.runtime/lib64/bionic/libc.so (ioctl+156)
    runtime.cc:655]   native: #02 pc 00000000000519e4  /system/lib64/libbinder.so (android::IPCThreadState::talkWithDriver(bool)+296)
    runtime.cc:655]   native: #03 pc 0000000000051bd4  /system/lib64/libbinder.so (android::IPCThreadState::getAndExecuteCommand()+24)
    runtime.cc:655]   native: #04 pc 0000000000052490  /system/lib64/libbinder.so (android::IPCThreadState::joinThreadPool(bool)+60)
    runtime.cc:655]   native: #05 pc 0000000000078470  /system/lib64/libbinder.so (android::PoolThread::threadLoop()+24)
    runtime.cc:655]   native: #06 pc 00000000000154cc  /system/lib64/libutils.so (android::Thread::_threadLoop(void*)+260)
    runtime.cc:655]   native: #07 pc 00000000000a2654  /system/lib64/libandroid_runtime.so (android::AndroidRuntime::javaThreadShell(void*)+144)
    runtime.cc:655]   native: #08 pc 0000000000014d90  /system/lib64/libutils.so (thread_data_t::trampoline(thread_data_t const*)+412)
    runtime.cc:655]   native: #09 pc 00000000000af888  /apex/com.android.runtime/lib64/bionic/libc.so (__pthread_start(void*)+64)
    runtime.cc:655]   native: #10 pc 000000000004fd08  /apex/com.android.runtime/lib64/bionic/libc.so (__start_thread+64)
    runtime.cc:655]   (no managed stack frames)
    runtime.cc:655] 
    runtime.cc:655] "Profile Saver" prio=5 tid=17 Native
    runtime.cc:655]   | group="" sCount=1 dsCount=0 flags=1 obj=0x79400688 self=0x7206f25820
    runtime.cc:655]   | sysTid=21480 nice=9 cgrp=default sched=0/0 handle=0x702be98cc0
    runtime.cc:655]   | state=S schedstat=( 329324 3358282 9 ) utm=0 stm=0 core=4 HZ=100
    runtime.cc:655]   | stack=0x702bda1000-0x702bda3000 stackSize=995KB
    runtime.cc:655]   | held mutexes=
    runtime.cc:655]   native: #00 pc 000000000004add0  /apex/com.android.runtime/lib64/bionic/libc.so (syscall+32)
    runtime.cc:655]   native: #01 pc 00000000001afcf4  /apex/com.android.art/lib64/libart.so (art::ConditionVariable::TimedWait(art::Thread*, long, int)+172)
    runtime.cc:655]   native: #02 pc 0000000000353b24  /apex/com.android.art/lib64/libart.so (art::ProfileSaver::Run()+188)
    runtime.cc:655]   native: #03 pc 00000000003589d8  /apex/com.android.art/lib64/libart.so (art::ProfileSaver::RunProfileSaverThread(void*)+180)
    runtime.cc:655]   native: #04 pc 00000000000af888  /apex/com.android.runtime/lib64/bionic/libc.so (__pthread_start(void*)+64)
    runtime.cc:655]   native: #05 pc 000000000004fd08  /apex/com.android.runtime/lib64/bionic/libc.so (__start_thread+64)
    runtime.cc:655]   (no managed stack frames)
    runtime.cc:655] 
    runtime.cc:655] "Thread-2" prio=4 tid=18 Waiting
    runtime.cc:655]   | group="" sCount=1 dsCount=0 flags=1 obj=0x79400700 self=0x7206f2e330
    runtime.cc:655]   | sysTid=21482 nice=10 cgrp=default sched=0/0 handle=0x702b521cc0
    runtime.cc:655]   | state=S schedstat=( 4497396 294844 5 ) utm=0 stm=0 core=7 HZ=100
    runtime.cc:655]   | stack=0x702b41e000-0x702b420000 stackSize=1043KB
    runtime.cc:655]   | held mutexes=
    runtime.cc:655]   native: #00 pc 000000000004adcc  /apex/com.android.runtime/lib64/bionic/libc.so (syscall+28)
    runtime.cc:655]   native: #01 pc 00000000005a2684  /apex/com.android.art/lib64/libart.so (art::Thread::Park(bool, long)+480)
    runtime.cc:655]   native: #02 pc 00000000004dbb3c  /apex/com.android.art/lib64/libart.so (art::Unsafe_park(_JNIEnv*, _jobject*, unsigned char, long)+660)
    runtime.cc:655]   at sun.misc.Unsafe.park(Native method)
    runtime.cc:655]   - waiting on an unknown object
    runtime.cc:655]   at java.util.concurrent.locks.LockSupport.park(LockSupport.java:190)
    runtime.cc:655]   at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.await(AbstractQueuedSynchronizer.java:2067)
    runtime.cc:655]   at java.util.concurrent.PriorityBlockingQueue.take(PriorityBlockingQueue.java:548)
    runtime.cc:655]   at com.android.volley.CacheDispatcher.processRequest(CacheDispatcher.java:114)
    runtime.cc:655]   at com.android.volley.CacheDispatcher.run(CacheDispatcher.java:93)
    runtime.cc:655] 
    runtime.cc:655] "Thread-3" prio=4 tid=19 Waiting
    runtime.cc:655]   | group="" sCount=1 dsCount=0 flags=1 obj=0x79400a18 self=0x7206f2ff00
    runtime.cc:655]   | sysTid=21483 nice=10 cgrp=default sched=0/0 handle=0x702b417cc0
    runtime.cc:655]   | state=S schedstat=( 44489532 1117134 25 ) utm=3 stm=1 core=6 HZ=100
    runtime.cc:655]   | stack=0x702b314000-0x702b316000 stackSize=1043KB
    runtime.cc:655]   | held mutexes=
    runtime.cc:655]   native: #00 pc 000000000004adcc  /apex/com.android.runtime/lib64/bionic/libc.so (syscall+28)
    runtime.cc:655]   native: #01 pc 00000000005a2684  /apex/com.android.art/lib64/libart.so (art::Thread::Park(bool, long)+480)
    runtime.cc:655]   native: #02 pc 00000000004dbb3c  /apex/com.android.art/lib64/libart.so (art::Unsafe_park(_JNIEnv*, _jobject*, unsigned char, long)+660)
    runtime.cc:655]   at sun.misc.Unsafe.park(Native method)
    runtime.cc:655]   - waiting on an unknown object
    runtime.cc:655]   at java.util.concurrent.locks.LockSupport.park(LockSupport.java:190)
    runtime.cc:655]   at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.await(AbstractQueuedSynchronizer.java:2067)
    runtime.cc:655]   at java.util.concurrent.PriorityBlockingQueue.take(PriorityBlockingQueue.java:548)
    runtime.cc:655]   at com.android.volley.NetworkDispatcher.processRequest(NetworkDispatcher.java:110)
    runtime.cc:655]   at com.android.volley.NetworkDispatcher.run(NetworkDispatcher.java:90)
    runtime.cc:655] 
    runtime.cc:655] "Thread-4" prio=4 tid=20 Native
    runtime.cc:655]   | group="" sCount=1 dsCount=0 flags=1 obj=0x79400d00 self=0x7206f2ab90
    runtime.cc:655]   | sysTid=21484 nice=10 cgrp=default sched=0/0 handle=0x702b30dcc0
    runtime.cc:655]   | state=S schedstat=( 3410364 0 3 ) utm=0 stm=0 core=4 HZ=100
    runtime.cc:655]   | stack=0x702b20a000-0x702b20c000 stackSize=1043KB
    runtime.cc:655]   | held mutexes=
    runtime.cc:655]   native: #00 pc 000000000009bb78  /apex/com.android.runtime/lib64/bionic/libc.so (__ppoll+8)
    runtime.cc:655]   native: #01 pc 0000000000059bc4  /apex/com.android.runtime/lib64/bionic/libc.so (poll+92)
    runtime.cc:655]   native: #02 pc 0000000000029138  /apex/com.android.art/lib64/libopenjdk.so (NET_Timeout+192)
    runtime.cc:655]   native: #03 pc 000000000002958c  /apex/com.android.art/lib64/libopenjdk.so (SocketInputStream_socketRead0+192)
    runtime.cc:655]   at java.net.SocketInputStream.socketRead0(Native method)
    runtime.cc:655]   at java.net.SocketInputStream.socketRead(SocketInputStream.java:119)
    runtime.cc:655]   at java.net.SocketInputStream.read(SocketInputStream.java:176)
    runtime.cc:655]   at java.net.SocketInputStream.read(SocketInputStream.java:144)
    runtime.cc:655]   at com.android.org.conscrypt.ConscryptEngineSocket$SSLInputStream.readFromSocket(ConscryptEngineSocket.java:928)
    runtime.cc:655]   at com.android.org.conscrypt.ConscryptEngineSocket$SSLInputStream.processDataFromSocket(ConscryptEngineSocket.java:892)
    runtime.cc:655]   at com.android.org.conscrypt.ConscryptEngineSocket$SSLInputStream.readUntilDataAvailable(ConscryptEngineSocket.java:807)
    runtime.cc:655]   at com.android.org.conscrypt.ConscryptEngineSocket$SSLInputStream.read(ConscryptEngineSocket.java:780)
    runtime.cc:655]   - locked <0x045b5ab0> (a java.lang.Object)
    runtime.cc:655]   at com.android.okhttp.okio.Okio$2.read(Okio.java:138)
    runtime.cc:655]   at com.android.okhttp.okio.AsyncTimeout$2.read(AsyncTimeout.java:213)
    runtime.cc:655]   at com.android.okhttp.okio.RealBufferedSource.indexOf(RealBufferedSource.java:307)
    runtime.cc:655]   at com.android.okhttp.okio.RealBufferedSource.indexOf(RealBufferedSource.java:301)
    runtime.cc:655]   at com.android.okhttp.okio.RealBufferedSource.readUtf8LineStrict(RealBufferedSource.java:197)
    runtime.cc:655]   at com.android.okhttp.internal.http.Http1xStream.readResponse(Http1xStream.java:188)
    runtime.cc:655]   at com.android.okhttp.internal.http.Http1xStream.readResponseHeaders(Http1xStream.java:129)
    runtime.cc:655]   at com.android.okhttp.internal.http.HttpEngine.readNetworkResponse(HttpEngine.java:750)
    runtime.cc:655]   at com.android.okhttp.internal.http.HttpEngine.readResponse(HttpEngine.java:622)
    runtime.cc:655]   at com.android.okhttp.internal.huc.HttpURLConnectionImpl.execute(HttpURLConnectionImpl.java:525)
    runtime.cc:655]   at com.android.okhttp.internal.huc.HttpURLConnectionImpl.getResponse(HttpURLConnectionImpl.java:461)
    runtime.cc:655]   at com.android.okhttp.internal.huc.HttpURLConnectionImpl.getResponseCode(HttpURLConnectionImpl.java:592)
    runtime.cc:655]   at com.android.okhttp.internal.huc.DelegatingHttpsURLConnection.getResponseCode(DelegatingHttpsURLConnection.java:106)
    runtime.cc:655]   at com.android.okhttp.internal.huc.HttpsURLConnectionImpl.getResponseCode(HttpsURLConnectionImpl.java:30)
    runtime.cc:655]   at com.android.volley.toolbox.HurlStack.executeRequest(HurlStack.java:91)
    runtime.cc:655]   at com.android.volley.toolbox.BasicNetwork.performRequest(BasicNetwork.java:104)
    runtime.cc:655]   at com.android.volley.NetworkDispatcher.processRequest(NetworkDispatcher.java:132)
    runtime.cc:655]   at com.android.volley.NetworkDispatcher.processRequest(NetworkDispatcher.java:111)
    runtime.cc:655]   at com.android.volley.NetworkDispatcher.run(NetworkDispatcher.java:90)
    runtime.cc:655] 
    runtime.cc:655] "Thread-6" prio=4 tid=21 Waiting
    runtime.cc:655]   | group="" sCount=1 dsCount=0 flags=1 obj=0x79400da8 self=0x7206f36e40
    runtime.cc:655]   | sysTid=21486 nice=10 cgrp=default sched=0/0 handle=0x702b0f9cc0
    runtime.cc:655]   | state=S schedstat=( 100365 0 1 ) utm=0 stm=0 core=6 HZ=100
    runtime.cc:655]   | stack=0x702aff6000-0x702aff8000 stackSize=1043KB
    runtime.cc:655]   | held mutexes=
    runtime.cc:655]   native: #00 pc 000000000004adcc  /apex/com.android.runtime/lib64/bionic/libc.so (syscall+28)
    runtime.cc:655]   native: #01 pc 00000000005a2684  /apex/com.android.art/lib64/libart.so (art::Thread::Park(bool, long)+480)
    runtime.cc:655]   native: #02 pc 00000000004dbb3c  /apex/com.android.art/lib64/libart.so (art::Unsafe_park(_JNIEnv*, _jobject*, unsigned char, long)+660)
    runtime.cc:655]   at sun.misc.Unsafe.park(Native method)
    runtime.cc:655]   - waiting on an unknown object
    runtime.cc:655]   at java.util.concurrent.locks.LockSupport.park(LockSupport.java:190)
    runtime.cc:655]   at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.await(AbstractQueuedSynchronizer.java:2067)
    runtime.cc:655]   at java.util.concurrent.PriorityBlockingQueue.take(PriorityBlockingQueue.java:548)
    runtime.cc:655]   at com.android.volley.NetworkDispatcher.processRequest(NetworkDispatcher.java:110)
    runtime.cc:655]   at com.android.volley.NetworkDispatcher.run(NetworkDispatcher.java:90)
    runtime.cc:655] 
    runtime.cc:655] "Thread-5" prio=4 tid=22 Waiting
    runtime.cc:655]   | group="" sCount=1 dsCount=0 flags=1 obj=0x79400e50 self=0x7206f336a0
    runtime.cc:655]   | sysTid=21485 nice=10 cgrp=default sched=0/0 handle=0x702b203cc0
    runtime.cc:655]   | state=S schedstat=( 151665 384844 4 ) utm=0 stm=0 core=6 HZ=100
    runtime.cc:655]   | stack=0x702b100000-0x702b102000 stackSize=1043KB
    runtime.cc:655]   | held mutexes=
    runtime.cc:655]   native: #00 pc 000000000004adcc  /apex/com.android.runtime/lib64/bionic/libc.so (syscall+28)
    runtime.cc:655]   native: #01 pc 00000000005a2684  /apex/com.android.art/lib64/libart.so (art::Thread::Park(bool, long)+480)
    runtime.cc:655]   native: #02 pc 00000000004dbb3c  /apex/com.android.art/lib64/libart.so (art::Unsafe_park(_JNIEnv*, _jobject*, unsigned char, long)+660)
    runtime.cc:655]   at sun.misc.Unsafe.park(Native method)
    runtime.cc:655]   - waiting on an unknown object
    runtime.cc:655]   at java.util.concurrent.locks.LockSupport.park(LockSupport.java:190)
    runtime.cc:655]   at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.await(AbstractQueuedSynchronizer.java:2067)
    runtime.cc:655]   at java.util.concurrent.PriorityBlockingQueue.take(PriorityBlockingQueue.java:548)
    runtime.cc:655]   at com.android.volley.NetworkDispatcher.processRequest(NetworkDispatcher.java:110)
    runtime.cc:655]   at com.android.volley.NetworkDispatcher.run(NetworkDispatcher.java:90)
    runtime.cc:655] 
    runtime.cc:655] "Thread-7" prio=10 tid=23 Native
    runtime.cc:655]   | group="" sCount=1 dsCount=0 flags=1 obj=0x79400ef8 self=0x7206f3a5e0
    runtime.cc:655]   | sysTid=21488 nice=-10 cgrp=default sched=0/0 handle=0x702bd9acc0
    runtime.cc:655]   | state=S schedstat=( 2022137 0 3 ) utm=0 stm=0 core=4 HZ=100
    runtime.cc:655]   | stack=0x702bca3000-0x702bca5000 stackSize=995KB
    runtime.cc:655]   | held mutexes=
    runtime.cc:655]   native: #00 pc 000000000004adcc  /apex/com.android.runtime/lib64/bionic/libc.so (syscall+28)
    runtime.cc:655]   native: #01 pc 000000000004e9b4  /apex/com.android.runtime/lib64/bionic/libc.so (__futex_wait_ex(void volatile*, bool, int, bool, timespec const*)+144)
    runtime.cc:655]   native: #02 pc 00000000000aec70  /apex/com.android.runtime/lib64/bionic/libc.so (pthread_cond_wait+60)
    runtime.cc:655]   native: #03 pc 00000000006f4994  /data/app/~~EljEbtMxoHPS3CC-1nKN_w==/org.dolphinemu.dolphinemu.debug-MTW6KXGos0F6setl2DoPwQ==/lib/arm64/libmain.so (std::__ndk1::condition_variable::wait(std::__ndk1::unique_lock<std::__ndk1::mutex>&)+20)
    runtime.cc:655]   native: #04 pc 00000000005f580c  /data/app/~~EljEbtMxoHPS3CC-1nKN_w==/org.dolphinemu.dolphinemu.debug-MTW6KXGos0F6setl2DoPwQ==/lib/arm64/libmain.so (???)
    runtime.cc:655]   native: #05 pc 00000000005f5ee4  /data/app/~~EljEbtMxoHPS3CC-1nKN_w==/org.dolphinemu.dolphinemu.debug-MTW6KXGos0F6setl2DoPwQ==/lib/arm64/libmain.so (???)
    runtime.cc:655]   native: #06 pc 00000000000af888  /apex/com.android.runtime/lib64/bionic/libc.so (__pthread_start(void*)+64)
    runtime.cc:655]   native: #07 pc 000000000004fd08  /apex/com.android.runtime/lib64/bionic/libc.so (__start_thread+64)
    runtime.cc:655]   (no managed stack frames)
    runtime.cc:655] 
    runtime.cc:655] "TcmReceiver" prio=5 tid=24 Native
    runtime.cc:655]   | group="" sCount=1 dsCount=0 flags=1 obj=0x79401148 self=0x7206f179a0
    runtime.cc:655]   | sysTid=21489 nice=0 cgrp=default sched=0/0 handle=0x702afefcc0
    runtime.cc:655]   | state=S schedstat=( 484323 52917 1 ) utm=0 stm=0 core=6 HZ=100
    runtime.cc:655]   | stack=0x702aeec000-0x702aeee000 stackSize=1043KB
    runtime.cc:655]   | held mutexes=
    runtime.cc:655]   native: #00 pc 000000000009b6b4  /apex/com.android.runtime/lib64/bionic/libc.so (__recvmsg+4)
    runtime.cc:655]   native: #01 pc 000000000005b008  /apex/com.android.runtime/lib64/bionic/libc.so (recvmsg+48)
    runtime.cc:655]   native: #02 pc 000000000000f7e0  /system/lib64/libbase.so (android::base::ReceiveFileDescriptorVector(android::base::borrowed_fd, void*, unsigned long, unsigned long, std::__1::vector<android::base::unique_fd_impl<android::base::DefaultCloser>, std::__1::allocator<android::base::unique_fd_impl<android::base::DefaultCloser> > >*)+356)
    runtime.cc:655]   native: #03 pc 0000000000119b7c  /system/lib64/libandroid_runtime.so (android::socket_read_all(_JNIEnv*, _jobject*, int, void*, unsigned long)+88)
    runtime.cc:655]   native: #04 pc 0000000000119800  /system/lib64/libandroid_runtime.so (android::socket_readba(_JNIEnv*, _jobject*, _jbyteArray*, int, int, _jobject*)+256)
    runtime.cc:655]   at android.net.LocalSocketImpl.readba_native(Native method)
    runtime.cc:655]   at android.net.LocalSocketImpl.access$300(LocalSocketImpl.java:37)
    runtime.cc:655]   at android.net.LocalSocketImpl$SocketInputStream.read(LocalSocketImpl.java:113)
    runtime.cc:655]   - locked <0x04a76d29> (a java.lang.Object)
    runtime.cc:655]   at com.qti.tcmclient.DpmTcmClient$TcmReceiver.run(DpmTcmClient.java:149)
    runtime.cc:655]   at java.lang.Thread.run(Thread.java:923)
    runtime.cc:655] 
    runtime.cc:655] "RenderThread" prio=10 tid=25 Native
    runtime.cc:655]   | group="" sCount=1 dsCount=0 flags=1 obj=0x79401268 self=0x7206f35270
    runtime.cc:655]   | sysTid=21490 nice=-10 cgrp=default sched=0/0 handle=0x702aee5cc0
    runtime.cc:655]   | state=S schedstat=( 102756414 48564252 497 ) utm=4 stm=5 core=5 HZ=100
    runtime.cc:655]   | stack=0x702adee000-0x702adf0000 stackSize=995KB
    runtime.cc:655]   | held mutexes=
    runtime.cc:655]   native: #00 pc 000000000009ba78  /apex/com.android.runtime/lib64/bionic/libc.so (__epoll_pwait+8)
    runtime.cc:655]   native: #01 pc 0000000000019acc  /system/lib64/libutils.so (android::Looper::pollInner(int)+184)
    runtime.cc:655]   native: #02 pc 00000000000199ac  /system/lib64/libutils.so (android::Looper::pollOnce(int, int*, int*, void**)+112)
    runtime.cc:655]   native: #03 pc 000000000020ec68  /system/lib64/libhwui.so (android::uirenderer::ThreadBase::waitForWork()+132)
    runtime.cc:655]   native: #04 pc 0000000000230064  /system/lib64/libhwui.so (android::uirenderer::renderthread::RenderThread::threadLoop()+80)
    runtime.cc:655]   native: #05 pc 00000000000154cc  /system/lib64/libutils.so (android::Thread::_threadLoop(void*)+260)
    runtime.cc:655]   native: #06 pc 0000000000014d90  /system/lib64/libutils.so (thread_data_t::trampoline(thread_data_t const*)+412)
    runtime.cc:655]   native: #07 pc 00000000000af888  /apex/com.android.runtime/lib64/bionic/libc.so (__pthread_start(void*)+64)
    runtime.cc:655]   native: #08 pc 000000000004fd08  /apex/com.android.runtime/lib64/bionic/libc.so (__start_thread+64)
    runtime.cc:655]   (no managed stack frames)
    runtime.cc:655] 
    runtime.cc:655] "OkHttp ConnectionPool" prio=5 tid=26 TimedWaiting
    runtime.cc:655]   | group="" sCount=1 dsCount=0 flags=1 obj=0x794012e0 self=0x7206f3c1b0
    runtime.cc:655]   | sysTid=21493 nice=0 cgrp=default sched=0/0 handle=0x702ade7cc0
    runtime.cc:655]   | state=S schedstat=( 398697 0 4 ) utm=0 stm=0 core=5 HZ=100
    runtime.cc:655]   | stack=0x702ace4000-0x702ace6000 stackSize=1043KB
    runtime.cc:655]   | held mutexes=
    runtime.cc:655]   native: #00 pc 000000000004add0  /apex/com.android.runtime/lib64/bionic/libc.so (syscall+32)
    runtime.cc:655]   native: #01 pc 00000000001afcf4  /apex/com.android.art/lib64/libart.so (art::ConditionVariable::TimedWait(art::Thread*, long, int)+172)
    runtime.cc:655]   native: #02 pc 000000000049a174  /apex/com.android.art/lib64/libart.so (art::Monitor::Wait(art::Thread*, long, int, bool, art::ThreadState)+564)
    runtime.cc:655]   native: #03 pc 000000000049b9b4  /apex/com.android.art/lib64/libart.so (art::Monitor::Wait(art::Thread*, art::ObjPtr<art::mirror::Object>, long, int, bool, art::ThreadState)+280)
    runtime.cc:655]   at java.lang.Object.wait(Native method)
    runtime.cc:655]   - waiting on <0x054228ae> (a com.android.okhttp.ConnectionPool)
    runtime.cc:655]   at com.android.okhttp.ConnectionPool$1.run(ConnectionPool.java:107)
    runtime.cc:655]   - locked <0x054228ae> (a com.android.okhttp.ConnectionPool)
    runtime.cc:655]   at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
    runtime.cc:655]   at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
    runtime.cc:655]   at java.lang.Thread.run(Thread.java:923)
    runtime.cc:655] 
    runtime.cc:655] "queued-work-looper" prio=6 tid=28 Native
    runtime.cc:655]   | group="" sCount=1 dsCount=0 flags=1 obj=0x79401408 self=0x7206f41520
    runtime.cc:655]   | sysTid=21496 nice=-2 cgrp=default sched=0/0 handle=0x701d702cc0
    runtime.cc:655]   | state=S schedstat=( 391041 483385 4 ) utm=0 stm=0 core=5 HZ=100
    runtime.cc:655]   | stack=0x701d5ff000-0x701d601000 stackSize=1043KB
    runtime.cc:655]   | held mutexes=
    runtime.cc:655]   native: #00 pc 000000000009ba78  /apex/com.android.runtime/lib64/bionic/libc.so (__epoll_pwait+8)
    runtime.cc:655]   native: #01 pc 0000000000019acc  /system/lib64/libutils.so (android::Looper::pollInner(int)+184)
    runtime.cc:655]   native: #02 pc 00000000000199ac  /system/lib64/libutils.so (android::Looper::pollOnce(int, int*, int*, void**)+112)
    runtime.cc:655]   native: #03 pc 0000000000112b04  /system/lib64/libandroid_runtime.so (android::android_os_MessageQueue_nativePollOnce(_JNIEnv*, _jobject*, long, int)+44)
    runtime.cc:655]   at android.os.MessageQueue.nativePollOnce(Native method)
    runtime.cc:655]   at android.os.MessageQueue.next(MessageQueue.java:335)
    runtime.cc:655]   at android.os.Looper.loop(Looper.java:206)
    runtime.cc:655]   at android.os.HandlerThread.run(HandlerThread.java:67)
    runtime.cc:655] 
    runtime.cc:655] "Picasso-Stats" prio=4 tid=29 Native
    runtime.cc:655]   | group="" sCount=1 dsCount=0 flags=1 obj=0x79401528 self=0x7206f38a10
    runtime.cc:655]   | sysTid=21497 nice=10 cgrp=default sched=0/0 handle=0x701c40ccc0
    runtime.cc:655]   | state=S schedstat=( 4542342 10003018 82 ) utm=0 stm=0 core=4 HZ=100
    runtime.cc:655]   | stack=0x701c309000-0x701c30b000 stackSize=1043KB
    runtime.cc:655]   | held mutexes=
    runtime.cc:655]   native: #00 pc 000000000009ba78  /apex/com.android.runtime/lib64/bionic/libc.so (__epoll_pwait+8)
    runtime.cc:655]   native: #01 pc 0000000000019acc  /system/lib64/libutils.so (android::Looper::pollInner(int)+184)
    runtime.cc:655]   native: #02 pc 00000000000199ac  /system/lib64/libutils.so (android::Looper::pollOnce(int, int*, int*, void**)+112)
    runtime.cc:655]   native: #03 pc 0000000000112b04  /system/lib64/libandroid_runtime.so (android::android_os_MessageQueue_nativePollOnce(_JNIEnv*, _jobject*, long, int)+44)
    runtime.cc:655]   at android.os.MessageQueue.nativePollOnce(Native method)
    runtime.cc:655]   at android.os.MessageQueue.next(MessageQueue.java:335)
    runtime.cc:655]   at android.os.Looper.loop(Looper.java:206)
    runtime.cc:655]   at android.os.HandlerThread.run(HandlerThread.java:67)
    runtime.cc:655] 
    runtime.cc:655] "Picasso-Dispatcher" prio=4 tid=30 Native
    runtime.cc:655]   | group="" sCount=1 dsCount=0 flags=1 obj=0x79401688 self=0x7206f3dd80
    runtime.cc:655]   | sysTid=21498 nice=10 cgrp=default sched=0/0 handle=0x701b302cc0
    runtime.cc:655]   | state=S schedstat=( 14865938 28821305 61 ) utm=0 stm=0 core=4 HZ=100
    runtime.cc:655]   | stack=0x701b1ff000-0x701b201000 stackSize=1043KB
    runtime.cc:655]   | held mutexes=
    runtime.cc:655]   native: #00 pc 000000000009ba78  /apex/com.android.runtime/lib64/bionic/libc.so (__epoll_pwait+8)
    runtime.cc:655]   native: #01 pc 0000000000019acc  /system/lib64/libutils.so (android::Looper::pollInner(int)+184)
    runtime.cc:655]   native: #02 pc 00000000000199ac  /system/lib64/libutils.so (android::Looper::pollOnce(int, int*, int*, void**)+112)
    runtime.cc:655]   native: #03 pc 0000000000112b04  /system/lib64/libandroid_runtime.so (android::android_os_MessageQueue_nativePollOnce(_JNIEnv*, _jobject*, long, int)+44)
    runtime.cc:655]   at android.os.MessageQueue.nativePollOnce(Native method)
    runtime.cc:655]   at android.os.MessageQueue.next(MessageQueue.java:335)
    runtime.cc:655]   at android.os.Looper.loop(Looper.java:206)
    runtime.cc:655]   at android.os.HandlerThread.run(HandlerThread.java:67)
    runtime.cc:655] 
    runtime.cc:655] "Picasso-refQueue" prio=4 tid=31 TimedWaiting
    runtime.cc:655]   | group="" sCount=1 dsCount=0 flags=1 obj=0x794017e8 self=0x7206f46890
    runtime.cc:655]   | sysTid=21499 nice=10 cgrp=default sched=0/0 handle=0x701a1f8cc0
    runtime.cc:655]   | state=S schedstat=( 349532 148124 4 ) utm=0 stm=0 core=5 HZ=100
    runtime.cc:655]   | stack=0x701a0f5000-0x701a0f7000 stackSize=1043KB
    runtime.cc:655]   | held mutexes=
    runtime.cc:655]   native: #00 pc 000000000004add0  /apex/com.android.runtime/lib64/bionic/libc.so (syscall+32)
    runtime.cc:655]   native: #01 pc 00000000001afcf4  /apex/com.android.art/lib64/libart.so (art::ConditionVariable::TimedWait(art::Thread*, long, int)+172)
    runtime.cc:655]   native: #02 pc 000000000049a174  /apex/com.android.art/lib64/libart.so (art::Monitor::Wait(art::Thread*, long, int, bool, art::ThreadState)+564)
    runtime.cc:655]   native: #03 pc 000000000049b9b4  /apex/com.android.art/lib64/libart.so (art::Monitor::Wait(art::Thread*, art::ObjPtr<art::mirror::Object>, long, int, bool, art::ThreadState)+280)
    runtime.cc:655]   at java.lang.Object.wait(Native method)
    runtime.cc:655]   - waiting on <0x0a24864f> (a java.lang.Object)
    runtime.cc:655]   at java.lang.Object.wait(Object.java:442)
    runtime.cc:655]   at java.lang.ref.ReferenceQueue.remove(ReferenceQueue.java:190)
    runtime.cc:655]   - locked <0x0a24864f> (a java.lang.Object)
    runtime.cc:655]   at com.squareup.picasso.Picasso$CleanupThread.run(Picasso.java:631)
    runtime.cc:655] 
    runtime.cc:655] "Okio Watchdog" prio=5 tid=32 TimedWaiting
    runtime.cc:655]   | group="" sCount=1 dsCount=0 flags=1 obj=0x79401a70 self=0x7206f48460
    runtime.cc:655]   | sysTid=21500 nice=0 cgrp=default sched=0/0 handle=0x70190eecc0
    runtime.cc:655]   | state=S schedstat=( 494999 151302 3 ) utm=0 stm=0 core=5 HZ=100
    runtime.cc:655]   | stack=0x7018feb000-0x7018fed000 stackSize=1043KB
    runtime.cc:655]   | held mutexes=
    runtime.cc:655]   native: #00 pc 000000000004add0  /apex/com.android.runtime/lib64/bionic/libc.so (syscall+32)
    runtime.cc:655]   native: #01 pc 00000000001afcf4  /apex/com.android.art/lib64/libart.so (art::ConditionVariable::TimedWait(art::Thread*, long, int)+172)
    runtime.cc:655]   native: #02 pc 000000000049a174  /apex/com.android.art/lib64/libart.so (art::Monitor::Wait(art::Thread*, long, int, bool, art::ThreadState)+564)
    runtime.cc:655]   native: #03 pc 000000000049b9b4  /apex/com.android.art/lib64/libart.so (art::Monitor::Wait(art::Thread*, art::ObjPtr<art::mirror::Object>, long, int, bool, art::ThreadState)+280)
    runtime.cc:655]   at java.lang.Object.wait(Native method)
    runtime.cc:655]   - waiting on <0x029371dc> (a java.lang.Class<com.android.okhttp.okio.AsyncTimeout>)
    runtime.cc:655]   at com.android.okhttp.okio.AsyncTimeout.awaitTimeout(AsyncTimeout.java:325)
    runtime.cc:655]   - locked <0x029371dc> (a java.lang.Class<com.android.okhttp.okio.AsyncTimeout>)
    runtime.cc:655]   at com.android.okhttp.okio.AsyncTimeout.access$000(AsyncTimeout.java:42)
    runtime.cc:655]   at com.android.okhttp.okio.AsyncTimeout$Watchdog.run(AsyncTimeout.java:288)
    runtime.cc:655] 
    runtime.cc:655] "Picasso-Idle" prio=4 tid=33 Waiting
    runtime.cc:655]   | group="" sCount=1 dsCount=0 flags=1 obj=0x79401b18 self=0x7206f22080
    runtime.cc:655]   | sysTid=21501 nice=10 cgrp=default sched=0/0 handle=0x7017fe4cc0
    runtime.cc:655]   | state=S schedstat=( 63987497 21993127 75 ) utm=6 stm=0 core=7 HZ=100
    runtime.cc:655]   | stack=0x7017ee1000-0x7017ee3000 stackSize=1043KB
    runtime.cc:655]   | held mutexes=
    runtime.cc:655]   native: #00 pc 000000000004adcc  /apex/com.android.runtime/lib64/bionic/libc.so (syscall+28)
    runtime.cc:655]   native: #01 pc 00000000005a2684  /apex/com.android.art/lib64/libart.so (art::Thread::Park(bool, long)+480)
    runtime.cc:655]   native: #02 pc 00000000004dbb3c  /apex/com.android.art/lib64/libart.so (art::Unsafe_park(_JNIEnv*, _jobject*, unsigned char, long)+660)
    runtime.cc:655]   at sun.misc.Unsafe.park(Native method)
    runtime.cc:655]   - waiting on an unknown object
    runtime.cc:655]   at java.util.concurrent.locks.LockSupport.park(LockSupport.java:190)
    runtime.cc:655]   at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.await(AbstractQueuedSynchronizer.java:2067)
    runtime.cc:655]   at java.util.concurrent.PriorityBlockingQueue.take(PriorityBlockingQueue.java:548)
    runtime.cc:655]   at java.util.concurrent.ThreadPoolExecutor.getTask(ThreadPoolExecutor.java:1092)
    runtime.cc:655]   at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1152)
    runtime.cc:655]   at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
    runtime.cc:655]   at java.lang.Thread.run(Thread.java:923)
    runtime.cc:655]   at com.squareup.picasso.Utils$PicassoThread.run(Utils.java:354)
    runtime.cc:655] 
    runtime.cc:655] "Picasso-Idle" prio=4 tid=34 Waiting
    runtime.cc:655]   | group="" sCount=1 dsCount=0 flags=1 obj=0x794054c0 self=0x7206f52b40
    runtime.cc:655]   | sysTid=21502 nice=10 cgrp=default sched=0/0 handle=0x7016edacc0
    runtime.cc:655]   | state=S schedstat=( 72291983 12236719 49 ) utm=6 stm=0 core=7 HZ=100
    runtime.cc:655]   | stack=0x7016dd7000-0x7016dd9000 stackSize=1043KB
    runtime.cc:655]   | held mutexes=
    runtime.cc:655]   native: #00 pc 000000000004adcc  /apex/com.android.runtime/lib64/bionic/libc.so (syscall+28)
    runtime.cc:655]   native: #01 pc 00000000005a2684  /apex/com.android.art/lib64/libart.so (art::Thread::Park(bool, long)+480)
    runtime.cc:655]   native: #02 pc 00000000004dbb3c  /apex/com.android.art/lib64/libart.so (art::Unsafe_park(_JNIEnv*, _jobject*, unsigned char, long)+660)
    runtime.cc:655]   at sun.misc.Unsafe.park(Native method)
    runtime.cc:655]   - waiting on an unknown object
    runtime.cc:655]   at java.util.concurrent.locks.LockSupport.park(LockSupport.java:190)
    runtime.cc:655]   at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.await(AbstractQueuedSynchronizer.java:2067)
    runtime.cc:655]   at java.util.concurrent.PriorityBlockingQueue.take(PriorityBlockingQueue.java:548)
    runtime.cc:655]   at java.util.concurrent.ThreadPoolExecutor.getTask(ThreadPoolExecutor.java:1092)
    runtime.cc:655]   at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1152)
    runtime.cc:655]   at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
    runtime.cc:655]   at java.lang.Thread.run(Thread.java:923)
    runtime.cc:655]   at com.squareup.picasso.Utils$PicassoThread.run(Utils.java:354)
    runtime.cc:655] 
    runtime.cc:655] "Picasso-Idle" prio=4 tid=35 Waiting
    runtime.cc:655]   | group="" sCount=1 dsCount=0 flags=1 obj=0x794055d8 self=0x7206f4f3a0
    runtime.cc:655]   | sysTid=21503 nice=10 cgrp=default sched=0/0 handle=0x7015dd0cc0
    runtime.cc:655]   | state=S schedstat=( 103046503 24082867 73 ) utm=9 stm=0 core=4 HZ=100
    runtime.cc:655]   | stack=0x7015ccd000-0x7015ccf000 stackSize=1043KB
    runtime.cc:655]   | held mutexes=
    runtime.cc:655]   native: #00 pc 000000000004adcc  /apex/com.android.runtime/lib64/bionic/libc.so (syscall+28)
    runtime.cc:655]   native: #01 pc 00000000005a2684  /apex/com.android.art/lib64/libart.so (art::Thread::Park(bool, long)+480)
    runtime.cc:655]   native: #02 pc 00000000004dbb3c  /apex/com.android.art/lib64/libart.so (art::Unsafe_park(_JNIEnv*, _jobject*, unsigned char, long)+660)
    runtime.cc:655]   at sun.misc.Unsafe.park(Native method)
    runtime.cc:655]   - waiting on an unknown object
    runtime.cc:655]   at java.util.concurrent.locks.LockSupport.park(LockSupport.java:190)
    runtime.cc:655]   at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.await(AbstractQueuedSynchronizer.java:2067)
    runtime.cc:655]   at java.util.concurrent.PriorityBlockingQueue.take(PriorityBlockingQueue.java:548)
    runtime.cc:655]   at java.util.concurrent.ThreadPoolExecutor.getTask(ThreadPoolExecutor.java:1092)
    runtime.cc:655]   at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1152)
    runtime.cc:655]   at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
    runtime.cc:655]   at java.lang.Thread.run(Thread.java:923)
    runtime.cc:655]   at com.squareup.picasso.Utils$PicassoThread.run(Utils.java:354)
    runtime.cc:655] 
    runtime.cc:655] "OkHttp ConnectionPool" prio=5 tid=27 TimedWaiting
    runtime.cc:655]   | group="" sCount=1 dsCount=0 flags=1 obj=0x794056a0 self=0x7206f430f0
    runtime.cc:655]   | sysTid=21508 nice=0 cgrp=default sched=0/0 handle=0x701e80ccc0
    runtime.cc:655]   | state=S schedstat=( 429636 0 4 ) utm=0 stm=0 core=4 HZ=100
    runtime.cc:655]   | stack=0x701e709000-0x701e70b000 stackSize=1043KB
    runtime.cc:655]   | held mutexes=
    runtime.cc:655]   native: #00 pc 000000000004add0  /apex/com.android.runtime/lib64/bionic/libc.so (syscall+32)
    runtime.cc:655]   native: #01 pc 00000000001afcf4  /apex/com.android.art/lib64/libart.so (art::ConditionVariable::TimedWait(art::Thread*, long, int)+172)
    runtime.cc:655]   native: #02 pc 000000000049a174  /apex/com.android.art/lib64/libart.so (art::Monitor::Wait(art::Thread*, long, int, bool, art::ThreadState)+564)
    runtime.cc:655]   native: #03 pc 000000000049b9b4  /apex/com.android.art/lib64/libart.so (art::Monitor::Wait(art::Thread*, art::ObjPtr<art::mirror::Object>, long, int, bool, art::ThreadState)+280)
    runtime.cc:655]   at java.lang.Object.wait(Native method)
    runtime.cc:655]   - waiting on <0x01a0b2e5> (a okhttp3.ConnectionPool)
    runtime.cc:655]   at okhttp3.ConnectionPool$1.run(ConnectionPool.java:67)
    runtime.cc:655]   - locked <0x01a0b2e5> (a okhttp3.ConnectionPool)
    runtime.cc:655]   at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
    runtime.cc:655]   at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
    runtime.cc:655]   at java.lang.Thread.run(Thread.java:923)
    runtime.cc:655] 
    runtime.cc:655] "Okio Watchdog" prio=5 tid=36 TimedWaiting
    runtime.cc:655]   | group="" sCount=1 dsCount=0 flags=1 obj=0x794058b0 self=0x7206f3f950
    runtime.cc:655]   | sysTid=21509 nice=0 cgrp=default sched=0/0 handle=0x70132b4cc0
    runtime.cc:655]   | state=S schedstat=( 313436 61563 10 ) utm=0 stm=0 core=6 HZ=100
    runtime.cc:655]   | stack=0x70131b1000-0x70131b3000 stackSize=1043KB
    runtime.cc:655]   | held mutexes=
    runtime.cc:655]   native: #00 pc 000000000004add0  /apex/com.android.runtime/lib64/bionic/libc.so (syscall+32)
    runtime.cc:655]   native: #01 pc 00000000001afcf4  /apex/com.android.art/lib64/libart.so (art::ConditionVariable::TimedWait(art::Thread*, long, int)+172)
    runtime.cc:655]   native: #02 pc 000000000049a174  /apex/com.android.art/lib64/libart.so (art::Monitor::Wait(art::Thread*, long, int, bool, art::ThreadState)+564)
    runtime.cc:655]   native: #03 pc 000000000049b9b4  /apex/com.android.art/lib64/libart.so (art::Monitor::Wait(art::Thread*, art::ObjPtr<art::mirror::Object>, long, int, bool, art::ThreadState)+280)
    runtime.cc:655]   at java.lang.Object.wait(Native method)
    runtime.cc:655]   - waiting on <0x0a0c21ba> (a java.lang.Class<okio.AsyncTimeout>)
    runtime.cc:655]   at okio.AsyncTimeout.awaitTimeout(AsyncTimeout.java:361)
    runtime.cc:655]   at okio.AsyncTimeout$Watchdog.run(AsyncTimeout.java:312)
    runtime.cc:655]   - locked <0x0a0c21ba> (a java.lang.Class<okio.AsyncTimeout>)
    runtime.cc:655] 
    runtime.cc:655] "NativeEmulation" prio=5 tid=37 Native
    runtime.cc:655]   | group="" sCount=1 dsCount=0 flags=1 obj=0x771680f0 self=0x7206f5edf0
    runtime.cc:655]   | sysTid=21515 nice=0 cgrp=default sched=0/0 handle=0x701209ccc0
    runtime.cc:655]   | state=S schedstat=( 16845103 17215256 97 ) utm=1 stm=0 core=6 HZ=100
    runtime.cc:655]   | stack=0x7011f99000-0x7011f9b000 stackSize=1043KB
    runtime.cc:655]   | held mutexes=
    runtime.cc:655]   native: #00 pc 000000000004adcc  /apex/com.android.runtime/lib64/bionic/libc.so (syscall+28)
    runtime.cc:655]   native: #01 pc 000000000004e9b4  /apex/com.android.runtime/lib64/bionic/libc.so (__futex_wait_ex(void volatile*, bool, int, bool, timespec const*)+144)
    runtime.cc:655]   native: #02 pc 00000000000aec70  /apex/com.android.runtime/lib64/bionic/libc.so (pthread_cond_wait+60)
    runtime.cc:655]   native: #03 pc 00000000006f4994  /data/app/~~EljEbtMxoHPS3CC-1nKN_w==/org.dolphinemu.dolphinemu.debug-MTW6KXGos0F6setl2DoPwQ==/lib/arm64/libmain.so (std::__ndk1::condition_variable::wait(std::__ndk1::unique_lock<std::__ndk1::mutex>&)+20)
    runtime.cc:655]   native: #04 pc 0000000000103650  /data/app/~~EljEbtMxoHPS3CC-1nKN_w==/org.dolphinemu.dolphinemu.debug-MTW6KXGos0F6setl2DoPwQ==/lib/arm64/libmain.so (???)
    runtime.cc:655]   native: #05 pc 0000000000103220  /data/app/~~EljEbtMxoHPS3CC-1nKN_w==/org.dolphinemu.dolphinemu.debug-MTW6KXGos0F6setl2DoPwQ==/lib/arm64/libmain.so (Java_org_dolphinemu_dolphinemu_NativeLibrary_Run___3Ljava_lang_String_2+64)
    runtime.cc:655]   native: #06 pc 000000000013ced4  /apex/com.android.art/lib64/libart.so (art_quick_generic_jni_trampoline+148)
    runtime.cc:655]   native: #07 pc 00000000001337e8  /apex/com.android.art/lib64/libart.so (art_quick_invoke_static_stub+568)
    runtime.cc:655]   native: #08 pc 00000000001a8a94  /apex/com.android.art/lib64/libart.so (art::ArtMethod::Invoke(art::Thread*, unsigned int*, unsigned int, art::JValue*, char const*)+228)
    runtime.cc:655]   native: #09 pc 0000000000318304  /apex/com.android.art/lib64/libart.so (art::interpreter::ArtInterpreterToCompiledCodeBridge(art::Thread*, art::ArtMethod*, art::ShadowFrame*, unsigned short, art::JValue*)+376)
    runtime.cc:655]   native: #10 pc 000000000030e630  /apex/com.android.art/lib64/libart.so (bool art::interpreter::DoCall<false, false>(art::ArtMethod*, art::Thread*, art::ShadowFrame&, art::Instruction const*, unsigned short, art::JValue*)+996)
    runtime.cc:655]   native: #11 pc 000000000067f514  /apex/com.android.art/lib64/libart.so (MterpInvokeStatic+548)
    runtime.cc:655]   native: #12 pc 000000000012d994  /apex/com.android.art/lib64/libart.so (mterp_op_invoke_static+20)
    runtime.cc:655]   native: #13 pc 0000000000005c18  [anon:dalvik-classes13.dex extracted in memory from /data/app/~~EljEbtMxoHPS3CC-1nKN_w==/org.dolphinemu.dolphinemu.debug-MTW6KXGos0F6setl2DoPwQ==/base.apk!classes13.dex] (org.dolphinemu.dolphinemu.fragments.EmulationFragment$EmulationState.lambda$runWithValidSurface$0$EmulationFragment$EmulationState+60)
    runtime.cc:655]   native: #14 pc 000000000067c85c  /apex/com.android.art/lib64/libart.so (MterpInvokeVirtual+1520)
    runtime.cc:655]   native: #15 pc 000000000012d814  /apex/com.android.art/lib64/libart.so (mterp_op_invoke_virtual+20)
    runtime.cc:655]   native: #16 pc 00000000000048f0  [anon:dalvik-classes13.dex extracted in memory from /data/app/~~EljEbtMxoHPS3CC-1nKN_w==/org.dolphinemu.dolphinemu.debug-MTW6KXGos0F6setl2DoPwQ==/base.apk!classes13.dex] (org.dolphinemu.dolphinemu.fragments.-$$Lambda$EmulationFragment$EmulationState$sj6x0Tho_him1soofrSKU6nojyY.run+4)
    runtime.cc:655]   native: #17 pc 000000000067e294  /apex/com.android.art/lib64/libart.so (MterpInvokeInterface+1808)
    runtime.cc:655]   native: #18 pc 000000000012da14  /apex/com.android.art/lib64/libart.so (mterp_op_invoke_interface+20)
    runtime.cc:655]   native: #19 pc 0000000000003330  [anon:dalvik-/apex/com.android.art/javalib/core-oj.jar-transformed] (java.lang.Thread.run+8)
    runtime.cc:655]   native: #20 pc 0000000000305c2c  /apex/com.android.art/lib64/libart.so (art::interpreter::Execute(art::Thread*, art::CodeItemDataAccessor const&, art::ShadowFrame&, art::JValue, bool, bool) (.llvm.17742421990365963658)+268)
    runtime.cc:655]   native: #21 pc 000000000066b2c4  /apex/com.android.art/lib64/libart.so (artQuickToInterpreterBridge+780)
    runtime.cc:655]   native: #22 pc 000000000013cff8  /apex/com.android.art/lib64/libart.so (art_quick_to_interpreter_bridge+88)
    runtime.cc:655]   native: #23 pc 0000000000133564  /apex/com.android.art/lib64/libart.so (art_quick_invoke_stub+548)
    runtime.cc:655]   native: #24 pc 00000000001a8a78  /apex/com.android.art/lib64/libart.so (art::ArtMethod::Invoke(art::Thread*, unsigned int*, unsigned int, art::JValue*, char const*)+200)
    runtime.cc:655]   native: #25 pc 0000000000554d80  /apex/com.android.art/lib64/libart.so (art::JValue art::InvokeVirtualOrInterfaceWithJValues<art::ArtMethod*>(art::ScopedObjectAccessAlreadyRunnable const&, _jobject*, art::ArtMethod*, jvalue const*)+460)
    runtime.cc:655]   native: #26 pc 00000000005a411c  /apex/com.android.art/lib64/libart.so (art::Thread::CreateCallback(void*)+1308)
    runtime.cc:655]   native: #27 pc 00000000000af888  /apex/com.android.runtime/lib64/bionic/libc.so (__pthread_start(void*)+64)
    runtime.cc:655]   native: #28 pc 000000000004fd08  /apex/com.android.runtime/lib64/bionic/libc.so (__start_thread+64)
    runtime.cc:655]   at org.dolphinemu.dolphinemu.NativeLibrary.Run(Native method)
    runtime.cc:655]   at org.dolphinemu.dolphinemu.fragments.EmulationFragment$EmulationState.lambda$runWithValidSurface$0$EmulationFragment$EmulationState(EmulationFragment.java:358)
    runtime.cc:655]   at org.dolphinemu.dolphinemu.fragments.-$$Lambda$EmulationFragment$EmulationState$sj6x0Tho_him1soofrSKU6nojyY.run(lambda:-1)
    runtime.cc:655]   at java.lang.Thread.run(Thread.java:923)
    runtime.cc:655] 
    runtime.cc:655] "Binder:21453_4" prio=5 tid=38 Native
    runtime.cc:655]   | group="" sCount=1 dsCount=0 flags=1 obj=0x77280020 self=0x7206f5d220
    runtime.cc:655]   | sysTid=21516 nice=0 cgrp=default sched=0/0 handle=0x7010f92cc0
    runtime.cc:655]   | state=S schedstat=( 2471411 1922605 13 ) utm=0 stm=0 core=6 HZ=100
    runtime.cc:655]   | stack=0x7010e9b000-0x7010e9d000 stackSize=995KB
    runtime.cc:655]   | held mutexes=
    runtime.cc:655]   native: #00 pc 000000000009aab4  /apex/com.android.runtime/lib64/bionic/libc.so (__ioctl+4)
    runtime.cc:655]   native: #01 pc 00000000000575e4  /apex/com.android.runtime/lib64/bionic/libc.so (ioctl+156)
    runtime.cc:655]   native: #02 pc 00000000000519e4  /system/lib64/libbinder.so (android::IPCThreadState::talkWithDriver(bool)+296)
    runtime.cc:655]   native: #03 pc 0000000000051bd4  /system/lib64/libbinder.so (android::IPCThreadState::getAndExecuteCommand()+24)
    runtime.cc:655]   native: #04 pc 0000000000052490  /system/lib64/libbinder.so (android::IPCThreadState::joinThreadPool(bool)+60)
    runtime.cc:655]   native: #05 pc 0000000000078470  /system/lib64/libbinder.so (android::PoolThread::threadLoop()+24)
    runtime.cc:655]   native: #06 pc 00000000000154cc  /system/lib64/libutils.so (android::Thread::_threadLoop(void*)+260)
    runtime.cc:655]   native: #07 pc 00000000000a2654  /system/lib64/libandroid_runtime.so (android::AndroidRuntime::javaThreadShell(void*)+144)
    runtime.cc:655]   native: #08 pc 0000000000014d90  /system/lib64/libutils.so (thread_data_t::trampoline(thread_data_t const*)+412)
    runtime.cc:655]   native: #09 pc 00000000000af888  /apex/com.android.runtime/lib64/bionic/libc.so (__pthread_start(void*)+64)
    runtime.cc:655]   native: #10 pc 000000000004fd08  /apex/com.android.runtime/lib64/bionic/libc.so (__start_thread+64)
    runtime.cc:655]   (no managed stack frames)
    runtime.cc:655] 
    runtime.cc:655] "hwuiTask1" prio=6 tid=39 Native
    runtime.cc:655]   | group="" sCount=1 dsCount=0 flags=1 obj=0x772c0020 self=0x7206f204b0
    runtime.cc:655]   | sysTid=21518 nice=-2 cgrp=default sched=0/0 handle=0x700fd96cc0
    runtime.cc:655]   | state=S schedstat=( 390468 2568959 2 ) utm=0 stm=0 core=5 HZ=100
    runtime.cc:655]   | stack=0x700fc9f000-0x700fca1000 stackSize=995KB
    runtime.cc:655]   | held mutexes=
    runtime.cc:655]   native: #00 pc 000000000004adcc  /apex/com.android.runtime/lib64/bionic/libc.so (syscall+28)
    runtime.cc:655]   native: #01 pc 000000000004e9b4  /apex/com.android.runtime/lib64/bionic/libc.so (__futex_wait_ex(void volatile*, bool, int, bool, timespec const*)+144)
    runtime.cc:655]   native: #02 pc 00000000000aec70  /apex/com.android.runtime/lib64/bionic/libc.so (pthread_cond_wait+60)
    runtime.cc:655]   native: #03 pc 00000000000584ec  /system/lib64/libc++.so (std::__1::condition_variable::wait(std::__1::unique_lock<std::__1::mutex>&)+20)
    runtime.cc:655]   native: #04 pc 00000000002339d0  /system/lib64/libhwui.so (android::uirenderer::CommonPool::workerLoop()+84)
    runtime.cc:655]   native: #05 pc 0000000000233bf8  /system/lib64/libhwui.so (void* std::__1::__thread_proxy<std::__1::tuple<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct> >, android::uirenderer::CommonPool::CommonPool()::$_0> >(void*)+128)
    runtime.cc:655]   native: #06 pc 00000000000af888  /apex/com.android.runtime/lib64/bionic/libc.so (__pthread_start(void*)+64)
    runtime.cc:655]   native: #07 pc 000000000004fd08  /apex/com.android.runtime/lib64/bionic/libc.so (__start_thread+64)
    runtime.cc:655]   (no managed stack frames)
    runtime.cc:655] 
    runtime.cc:655] "hwuiTask0" prio=6 tid=40 Native
    runtime.cc:655]   | group="" sCount=1 dsCount=0 flags=1 obj=0x77300020 self=0x7206f609c0
    runtime.cc:655]   | sysTid=21517 nice=-2 cgrp=default sched=0/0 handle=0x700fe94cc0
    runtime.cc:655]   | state=S schedstat=( 267292 2761250 2 ) utm=0 stm=0 core=5 HZ=100
    runtime.cc:655]   | stack=0x700fd9d000-0x700fd9f000 stackSize=995KB
    runtime.cc:655]   | held mutexes=
    runtime.cc:655]   native: #00 pc 000000000004adcc  /apex/com.android.runtime/lib64/bionic/libc.so (syscall+28)
    runtime.cc:655]   native: #01 pc 000000000004e9b4  /apex/com.android.runtime/lib64/bionic/libc.so (__futex_wait_ex(void volatile*, bool, int, bool, timespec const*)+144)
    runtime.cc:655]   native: #02 pc 00000000000aec70  /apex/com.android.runtime/lib64/bionic/libc.so (pthread_cond_wait+60)
    runtime.cc:655]   native: #03 pc 00000000000584ec  /system/lib64/libc++.so (std::__1::condition_variable::wait(std::__1::unique_lock<std::__1::mutex>&)+20)
    runtime.cc:655]   native: #04 pc 00000000002339d0  /system/lib64/libhwui.so (android::uirenderer::CommonPool::workerLoop()+84)
    runtime.cc:655]   native: #05 pc 0000000000233bf8  /system/lib64/libhwui.so (void* std::__1::__thread_proxy<std::__1::tuple<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct> >, android::uirenderer::CommonPool::CommonPool()::$_0> >(void*)+128)
    runtime.cc:655]   native: #06 pc 00000000000af888  /apex/com.android.runtime/lib64/bionic/libc.so (__pthread_start(void*)+64)
    runtime.cc:655]   native: #07 pc 000000000004fd08  /apex/com.android.runtime/lib64/bionic/libc.so (__start_thread+64)
    runtime.cc:655]   (no managed stack frames)
    runtime.cc:655] 
    runtime.cc:655] "InsetsAnimations" prio=5 tid=41 Native
    runtime.cc:655]   | group="" sCount=1 dsCount=0 flags=1 obj=0x7716be78 self=0x7206f64160
    runtime.cc:655]   | sysTid=21519 nice=0 cgrp=default sched=0/0 handle=0x700dc35cc0
    runtime.cc:655]   | state=S schedstat=( 19900108 29030198 74 ) utm=1 stm=0 core=4 HZ=100
    runtime.cc:655]   | stack=0x700db32000-0x700db34000 stackSize=1043KB
    runtime.cc:655]   | held mutexes=
    runtime.cc:655]   native: #00 pc 000000000009ba78  /apex/com.android.runtime/lib64/bionic/libc.so (__epoll_pwait+8)
    runtime.cc:655]   native: #01 pc 0000000000019acc  /system/lib64/libutils.so (android::Looper::pollInner(int)+184)
    runtime.cc:655]   native: #02 pc 00000000000199ac  /system/lib64/libutils.so (android::Looper::pollOnce(int, int*, int*, void**)+112)
    runtime.cc:655]   native: #03 pc 0000000000112b04  /system/lib64/libandroid_runtime.so (android::android_os_MessageQueue_nativePollOnce(_JNIEnv*, _jobject*, long, int)+44)
    runtime.cc:655]   at android.os.MessageQueue.nativePollOnce(Native method)
    runtime.cc:655]   at android.os.MessageQueue.next(MessageQueue.java:335)
    runtime.cc:655]   at android.os.Looper.loop(Looper.java:206)
    runtime.cc:655]   at android.os.HandlerThread.run(HandlerThread.java:67)
    runtime.cc:655] 
    runtime.cc:655] "AudioTrack" prio=10 tid=42 Native
    runtime.cc:655]   | group="" sCount=1 dsCount=0 flags=1 obj=0x773c0020 self=0x7206f6e840
    runtime.cc:655]   | sysTid=21522 nice=-16 cgrp=default sched=1073741825/2 handle=0x700b49acc0
    runtime.cc:655]   | state=S schedstat=( 17536767 890467 240 ) utm=1 stm=0 core=6 HZ=100
    runtime.cc:655]   | stack=0x700b3a3000-0x700b3a5000 stackSize=995KB
    runtime.cc:655]   | held mutexes=
    runtime.cc:655]   native: #00 pc 000000000004adcc  /apex/com.android.runtime/lib64/bionic/libc.so (syscall+28)
    runtime.cc:655]   native: #01 pc 0000000000081578  /system/lib64/libaudioclient.so (android::ClientProxy::obtainBuffer(android::Proxy::Buffer*, timespec const*, timespec*)+688)
    runtime.cc:655]   native: #02 pc 0000000000078cc0  /system/lib64/libaudioclient.so (android::AudioTrack::obtainBuffer(android::AudioTrack::Buffer*, timespec const*, timespec*, unsigned long*)+572)
    runtime.cc:655]   native: #03 pc 00000000000782ec  /system/lib64/libaudioclient.so (android::AudioTrack::processAudioBuffer()+2220)
    runtime.cc:655]   native: #04 pc 0000000000077750  /system/lib64/libaudioclient.so (android::AudioTrack::AudioTrackThread::threadLoop()+272)
    runtime.cc:655]   native: #05 pc 0000000000015594  /system/lib64/libutils.so (android::Thread::_threadLoop(void*)+460)
    runtime.cc:655]   native: #06 pc 00000000000a2654  /system/lib64/libandroid_runtime.so (android::AndroidRuntime::javaThreadShell(void*)+144)
    runtime.cc:655]   native: #07 pc 0000000000014d90  /system/lib64/libutils.so (thread_data_t::trampoline(thread_data_t const*)+412)
    runtime.cc:655]   native: #08 pc 00000000000af888  /apex/com.android.runtime/lib64/bionic/libc.so (__pthread_start(void*)+64)
    runtime.cc:655]   native: #09 pc 000000000004fd08  /apex/com.android.runtime/lib64/bionic/libc.so (__start_thread+64)
    runtime.cc:655]   (no managed stack frames)
    runtime.cc:655] 
    runtime.cc:655] Aborting thread:
    runtime.cc:655] "Thread-12" prio=5 tid=43 Native
    runtime.cc:655]   | group="" sCount=0 dsCount=0 flags=0 obj=0x77480000 self=0x720712b840
    runtime.cc:655]   | sysTid=21520 nice=0 cgrp=default sched=0/0 handle=0x700cb2bcc0
    runtime.cc:655]   | state=R schedstat=( 962309831 30070985 441 ) utm=79 stm=16 core=7 HZ=100
    runtime.cc:655]   | stack=0x700ca34000-0x700ca36000 stackSize=995KB
    runtime.cc:655]   | held mutexes= "abort lock"
    runtime.cc:655]   native: #00 pc 000000000049ef64  /apex/com.android.art/lib64/libart.so (art::DumpNativeStack(std::__1::basic_ostream<char, std::__1::char_traits<char> >&, int, BacktraceMap*, char const*, art::ArtMethod*, void*, bool)+140)
    runtime.cc:655]   native: #01 pc 00000000005ac0bc  /apex/com.android.art/lib64/libart.so (art::Thread::DumpStack(std::__1::basic_ostream<char, std::__1::char_traits<char> >&, bool, BacktraceMap*, bool) const+376)
    runtime.cc:655]   native: #02 pc 00000000005717ec  /apex/com.android.art/lib64/libart.so (art::AbortState::DumpThread(std::__1::basic_ostream<char, std::__1::char_traits<char> >&, art::Thread*) const+60)
    runtime.cc:655]   native: #03 pc 000000000055d1c4  /apex/com.android.art/lib64/libart.so (art::Runtime::Abort(char const*)+2436)
    runtime.cc:655]   native: #04 pc 0000000000013978  /system/lib64/libbase.so (android::base::SetAborter(std::__1::function<void (char const*)>&&)::$_3::__invoke(char const*)+76)
    runtime.cc:655]   native: #05 pc 0000000000012fa4  /system/lib64/libbase.so (android::base::LogMessage::~LogMessage()+320)
    runtime.cc:655]   native: #06 pc 00000000003851a0  /apex/com.android.art/lib64/libart.so (art::JavaVMExt::JniAbort(char const*, char const*)+2572)
    runtime.cc:655]   native: #07 pc 0000000000385218  /apex/com.android.art/lib64/libart.so (art::JavaVMExt::JniAbortV(char const*, char const*, std::__va_list)+108)
    runtime.cc:655]   native: #08 pc 0000000000376efc  /apex/com.android.art/lib64/libart.so (art::(anonymous namespace)::ScopedCheck::AbortF(char const*, ...)+144)
    runtime.cc:655]   native: #09 pc 0000000000375a14  /apex/com.android.art/lib64/libart.so (art::(anonymous namespace)::ScopedCheck::CheckPossibleHeapValue(art::ScopedObjectAccess&, char, art::(anonymous namespace)::JniValueType)+3108)
    runtime.cc:655]   native: #10 pc 0000000000374674  /apex/com.android.art/lib64/libart.so (art::(anonymous namespace)::ScopedCheck::Check(art::ScopedObjectAccess&, bool, char const*, art::(anonymous namespace)::JniValueType*)+612)
    runtime.cc:655]   native: #11 pc 0000000000369af0  /apex/com.android.art/lib64/libart.so (art::(anonymous namespace)::CheckJNI::NewString(_JNIEnv*, unsigned short const*, int)+700)
    runtime.cc:655]   native: #12 pc 0000000000104d04  /data/app/~~EljEbtMxoHPS3CC-1nKN_w==/org.dolphinemu.dolphinemu.debug-MTW6KXGos0F6setl2DoPwQ==/lib/arm64/libmain.so (???)
    runtime.cc:655]   native: #13 pc 0000000000102e74  /data/app/~~EljEbtMxoHPS3CC-1nKN_w==/org.dolphinemu.dolphinemu.debug-MTW6KXGos0F6setl2DoPwQ==/lib/arm64/libmain.so (???)
    runtime.cc:655]   native: #14 pc 00000000001179dc  /data/app/~~EljEbtMxoHPS3CC-1nKN_w==/org.dolphinemu.dolphinemu.debug-MTW6KXGos0F6setl2DoPwQ==/lib/arm64/libmain.so (???)
    runtime.cc:655]   native: #15 pc 00000000001f6430  /data/app/~~EljEbtMxoHPS3CC-1nKN_w==/org.dolphinemu.dolphinemu.debug-MTW6KXGos0F6setl2DoPwQ==/lib/arm64/libmain.so (???)
    runtime.cc:655]   native: #16 pc 0000000002000ea0  <anonymous:6ff9f06000> (???)
    runtime.cc:655]   (no managed stack frames)
    runtime.cc:655] Pending exception java.lang.StackOverflowError: stack size 995KB
    runtime.cc:655] (Throwable with empty stack trace)
    runtime.cc:655] 
    runtime.cc:663] JNI DETECTED ERROR IN APPLICATION: JNI NewString called with pending exception java.lang.StackOverflowError: stack size 995KB
    runtime.cc:663] (Throwable with empty stack trace)
    runtime.cc:663] 
    runtime.cc:663]     in call to NewString
A/libc: Fatal signal 6 (SIGABRT), code -1 (SI_QUEUE) in tid 21520 (Thread-12), pid 21453 (olphinemu.debug)
```
</details>